{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# metaBEAT analysis of Illumina MiSeq output for Filtration_12S_July_2016"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mark sure your metaBEAT image version is the latest\n",
    "\n",
    "`metaBEAT - metaBarcoding and Environmental DNA Analyses tool version: v.0.97.8-global`\n",
    "Update from your terminal\n",
    "\n",
    "####check the image, the latest ID should be 1f9e1e7d4566\n",
    "\n",
    "sudo docker images\n",
    "####if not, use the command to update the image from your terminal\n",
    "\n",
    "sudo docker pull chrishah/metabeat\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###The first step will be to trim/clean our raw Illumina data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/trimming\n"
     ]
    }
   ],
   "source": [
    "cd trimming"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Prepare Querymap__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Prepare a text file specifying the samples to be processed including the format and location of the reads. \n",
    "\n",
    "The below command expects the Illumina data to be present in 2 fastq files (forward and reverse reads) per sample in a directory `./raw_reads/`. It expects the files to be named 'sampleID', followed by '_R1' or '_R2' to identify the forward/reverse read file respectively. \n",
    "\n",
    "The raw data need to be downloaded with `How_to_download_Rawdata_from_SRA.ipynb` (see [here](https://github.com/HullUni-bioinformatics/Li_et_al_2017/blob/master/raw_reads/How_to_download_Rawdata_from_SRA.ipynb))\n",
    "\n",
    "SampleID must corresponds to the first column in the file `Sample_accessions.tsv` (see [here](https://github.com/HullUni-bioinformatics/Li_et_al_2017/blob/master/supplementary_data/Sample_accessions.tsv)), marker is '12S'.\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 277852\r\n",
      "-rw-r--r-- 1 root root 1169883 Oct 31 14:35 T1-1-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1080602 Oct 31 14:35 T1-1-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 5059798 Oct 31 14:35 T1-1-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 4350541 Oct 31 14:35 T1-1-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1646705 Oct 31 14:35 T1-1-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1524692 Oct 31 14:35 T1-1-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  997465 Oct 31 14:35 T1-1-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  915492 Oct 31 14:35 T1-1-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1264645 Oct 31 14:35 T1-1-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1119177 Oct 31 14:35 T1-1-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  923927 Oct 31 14:35 T1-2-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  896834 Oct 31 14:35 T1-2-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  874185 Oct 31 14:35 T1-2-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  809912 Oct 31 14:35 T1-2-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  696940 Oct 31 14:35 T1-2-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  686118 Oct 31 14:35 T1-2-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  559125 Oct 31 14:35 T1-2-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  530555 Oct 31 14:35 T1-2-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1141489 Oct 31 14:35 T1-2-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1062753 Oct 31 14:35 T1-2-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  863516 Oct 31 14:35 T1-3-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  737043 Oct 31 14:35 T1-3-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1754664 Oct 31 14:35 T1-3-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1520635 Oct 31 14:35 T1-3-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  712991 Oct 31 14:35 T1-3-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  644004 Oct 31 14:35 T1-3-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  760795 Oct 31 14:35 T1-3-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  739284 Oct 31 14:35 T1-3-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  627209 Oct 31 14:35 T1-3-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  623681 Oct 31 14:35 T1-3-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  305656 Oct 31 14:35 T1-4-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  268115 Oct 31 14:35 T1-4-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  906665 Oct 31 14:35 T1-4-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  813307 Oct 31 14:35 T1-4-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  353778 Oct 31 14:35 T1-4-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  323808 Oct 31 14:35 T1-4-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  799703 Oct 31 14:35 T1-4-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  806782 Oct 31 14:35 T1-4-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  596656 Oct 31 14:35 T1-4-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  591552 Oct 31 14:35 T1-4-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  260787 Oct 31 14:35 T2-1-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  244877 Oct 31 14:35 T2-1-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  514316 Oct 31 14:36 T2-1-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  462166 Oct 31 14:36 T2-1-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  273732 Oct 31 14:36 T2-1-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  277648 Oct 31 14:36 T2-1-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  510004 Oct 31 14:36 T2-1-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  486827 Oct 31 14:36 T2-1-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  405361 Oct 31 14:36 T2-1-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  352744 Oct 31 14:36 T2-1-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  521719 Oct 31 14:36 T2-2-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  514790 Oct 31 14:36 T2-2-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  778929 Oct 31 14:36 T2-2-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  729217 Oct 31 14:36 T2-2-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2539468 Oct 31 14:37 T2-2-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2545523 Oct 31 14:37 T2-2-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  916314 Oct 31 14:37 T2-2-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  880654 Oct 31 14:37 T2-2-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  803566 Oct 31 14:37 T2-2-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  733345 Oct 31 14:37 T2-2-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  782489 Oct 31 14:37 T2-3-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  776217 Oct 31 14:37 T2-3-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  529197 Oct 31 14:37 T2-3-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  538454 Oct 31 14:37 T2-3-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  400073 Oct 31 14:37 T2-3-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  380643 Oct 31 14:37 T2-3-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1987842 Oct 31 14:38 T2-3-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1750375 Oct 31 14:38 T2-3-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1251687 Oct 31 14:38 T2-3-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1085652 Oct 31 14:38 T2-3-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1515560 Oct 31 14:38 T2-4-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1519213 Oct 31 14:38 T2-4-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  801240 Oct 31 14:38 T2-4-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  797220 Oct 31 14:38 T2-4-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  532724 Oct 31 14:38 T2-4-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  514938 Oct 31 14:38 T2-4-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2124654 Oct 31 14:39 T2-4-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1829951 Oct 31 14:39 T2-4-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  720440 Oct 31 14:39 T2-4-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  640400 Oct 31 14:39 T2-4-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  837561 Oct 31 14:39 T3-1-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  797409 Oct 31 14:39 T3-1-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  321701 Oct 31 14:39 T3-1-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  302775 Oct 31 14:39 T3-1-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  782257 Oct 31 14:39 T3-1-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  729441 Oct 31 14:39 T3-1-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1849370 Oct 31 14:40 T3-1-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1799266 Oct 31 14:40 T3-1-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  607559 Oct 31 14:40 T3-1-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  559684 Oct 31 14:40 T3-1-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  707649 Oct 31 14:40 T3-2-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  687989 Oct 31 14:40 T3-2-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  842133 Oct 31 14:40 T3-2-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  786624 Oct 31 14:40 T3-2-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  871607 Oct 31 14:40 T3-2-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  803165 Oct 31 14:40 T3-2-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  393703 Oct 31 14:40 T3-2-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  375224 Oct 31 14:40 T3-2-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  635206 Oct 31 14:40 T3-2-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  604321 Oct 31 14:40 T3-2-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1517642 Oct 31 14:41 T3-3-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1399229 Oct 31 14:41 T3-3-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1182477 Oct 31 14:41 T3-3-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1152572 Oct 31 14:41 T3-3-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1482888 Oct 31 14:42 T3-3-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1474615 Oct 31 14:42 T3-3-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1016714 Oct 31 14:42 T3-3-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  995734 Oct 31 14:42 T3-3-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1593286 Oct 31 14:42 T3-3-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1570280 Oct 31 14:42 T3-3-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1017540 Oct 31 14:42 T3-4-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  916070 Oct 31 14:42 T3-4-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  662832 Oct 31 14:42 T3-4-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  647731 Oct 31 14:42 T3-4-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1130869 Oct 31 14:42 T3-4-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1128475 Oct 31 14:42 T3-4-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  273869 Oct 31 14:43 T3-4-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  263971 Oct 31 14:43 T3-4-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  514149 Oct 31 14:43 T3-4-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  512704 Oct 31 14:43 T3-4-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  812408 Oct 31 14:43 T4-1-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  766670 Oct 31 14:43 T4-1-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2997095 Oct 31 14:43 T4-1-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2661223 Oct 31 14:43 T4-1-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2622559 Oct 31 14:43 T4-1-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2386970 Oct 31 14:43 T4-1-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1737530 Oct 31 14:44 T4-1-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1577848 Oct 31 14:44 T4-1-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2770559 Oct 31 14:44 T4-1-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2494275 Oct 31 14:44 T4-1-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  862849 Oct 31 14:44 T4-2-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  836973 Oct 31 14:44 T4-2-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1292962 Oct 31 14:44 T4-2-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1175963 Oct 31 14:44 T4-2-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1263178 Oct 31 14:44 T4-2-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1205551 Oct 31 14:44 T4-2-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1199358 Oct 31 14:45 T4-2-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1132996 Oct 31 14:45 T4-2-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2328390 Oct 31 14:45 T4-2-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2221757 Oct 31 14:45 T4-2-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1394580 Oct 31 14:45 T4-3-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1236623 Oct 31 14:45 T4-3-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2824253 Oct 31 14:45 T4-3-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2525880 Oct 31 14:45 T4-3-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1413879 Oct 31 14:45 T4-3-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1298289 Oct 31 14:45 T4-3-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1157405 Oct 31 14:46 T4-3-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1183609 Oct 31 14:46 T4-3-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1835758 Oct 31 14:46 T4-3-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1888757 Oct 31 14:46 T4-3-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  548645 Oct 31 14:46 T4-4-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  484207 Oct 31 14:46 T4-4-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  960807 Oct 31 14:46 T4-4-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  850374 Oct 31 14:46 T4-4-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  877436 Oct 31 14:46 T4-4-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  774581 Oct 31 14:46 T4-4-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1841221 Oct 31 14:47 T4-4-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1801585 Oct 31 14:47 T4-4-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1973550 Oct 31 14:47 T4-4-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1999651 Oct 31 14:47 T4-4-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  273891 Oct 31 14:47 T5-1-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  259177 Oct 31 14:47 T5-1-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1177509 Oct 31 14:47 T5-1-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1062842 Oct 31 14:47 T5-1-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1243259 Oct 31 14:47 T5-1-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1200531 Oct 31 14:47 T5-1-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2356686 Oct 31 14:48 T5-1-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2185309 Oct 31 14:48 T5-1-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  655485 Oct 31 14:48 T5-1-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  590689 Oct 31 14:48 T5-1-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  412765 Oct 31 14:48 T5-2-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  393682 Oct 31 14:48 T5-2-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  735380 Oct 31 14:48 T5-2-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  667296 Oct 31 14:48 T5-2-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1276077 Oct 31 14:48 T5-2-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1273284 Oct 31 14:48 T5-2-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  349793 Oct 31 14:48 T5-2-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  322145 Oct 31 14:48 T5-2-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2017246 Oct 31 14:48 T5-2-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1775386 Oct 31 14:48 T5-2-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  255924 Oct 31 14:49 T5-3-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  263001 Oct 31 14:49 T5-3-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1409107 Oct 31 14:49 T5-3-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1369435 Oct 31 14:49 T5-3-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  759630 Oct 31 14:49 T5-3-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  721078 Oct 31 14:49 T5-3-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1292780 Oct 31 14:49 T5-3-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1160264 Oct 31 14:49 T5-3-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1071771 Oct 31 14:49 T5-3-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  970358 Oct 31 14:49 T5-3-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1376863 Oct 31 14:50 T5-4-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1364244 Oct 31 14:50 T5-4-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  909645 Oct 31 14:50 T5-4-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  920498 Oct 31 14:50 T5-4-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  539146 Oct 31 14:50 T5-4-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  532004 Oct 31 14:50 T5-4-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  590597 Oct 31 14:50 T5-4-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  507467 Oct 31 14:50 T5-4-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  652587 Oct 31 14:50 T5-4-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  577507 Oct 31 14:50 T5-4-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2315626 Oct 31 14:50 TP5-1-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2227292 Oct 31 14:50 TP5-1-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1569841 Oct 31 14:51 TP5-1-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1487541 Oct 31 14:51 TP5-1-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 4462825 Oct 31 14:51 TP5-1-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 3983072 Oct 31 14:51 TP5-1-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1314947 Oct 31 14:51 TP5-1-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1191652 Oct 31 14:51 TP5-1-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1755554 Oct 31 14:51 TP5-1-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1639320 Oct 31 14:51 TP5-1-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 3681934 Oct 31 14:52 TP5-2-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 3601031 Oct 31 14:52 TP5-2-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2451411 Oct 31 14:52 TP5-2-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2345735 Oct 31 14:52 TP5-2-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1857408 Oct 31 14:52 TP5-2-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1738540 Oct 31 14:52 TP5-2-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2775415 Oct 31 14:53 TP5-2-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2623868 Oct 31 14:53 TP5-2-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1167915 Oct 31 14:53 TP5-2-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1076231 Oct 31 14:53 TP5-2-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  913547 Oct 31 14:53 TP5-3-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  851300 Oct 31 14:53 TP5-3-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  574497 Oct 31 14:53 TP5-3-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  596983 Oct 31 14:53 TP5-3-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1123903 Oct 31 14:53 TP5-3-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1160689 Oct 31 14:53 TP5-3-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1087653 Oct 31 14:53 TP5-3-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1123792 Oct 31 14:53 TP5-3-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1246978 Oct 31 14:54 TP5-3-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1258955 Oct 31 14:54 TP5-3-5-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1577371 Oct 31 14:54 TP5-4-1-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1425860 Oct 31 14:54 TP5-4-1-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2268032 Oct 31 14:54 TP5-4-2-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 2284387 Oct 31 14:54 TP5-4-2-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  807706 Oct 31 14:54 TP5-4-3-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  849716 Oct 31 14:54 TP5-4-3-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  976759 Oct 31 14:54 TP5-4-4-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root  968012 Oct 31 14:54 TP5-4-4-12S_R2.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1970504 Oct 31 14:54 TP5-4-5-12S_R1.fastq.gz\r\n",
      "-rw-r--r-- 1 root root 1937722 Oct 31 14:54 TP5-4-5-12S_R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls -l ../raw_reads/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "%%bash\n",
    "\n",
    "for a in $(cat ../supplementary_data/Sample_accessions.tsv | grep \"12S\" | cut -f 1 | grep \"SampleID\" -v)\n",
    "do\n",
    "    R1=$(ls -1 ../raw_reads/$a-12S_* | grep \"_R1.fastq\")\n",
    "    R2=$(ls -1 ../raw_reads/$a-12S_* | grep \"_R2.fastq\")\n",
    "\n",
    "    echo -e \"$a\\tfastq\\t$R1\\t$R2\"\n",
    "done > Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The resulting file should look e.g. like below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "T1-1-1\tfastq\t../raw_reads/T1-1-1-12S_R1.fastq.gz\t../raw_reads/T1-1-1-12S_R2.fastq.gz\r\n",
      "T1-1-2\tfastq\t../raw_reads/T1-1-2-12S_R1.fastq.gz\t../raw_reads/T1-1-2-12S_R2.fastq.gz\r\n",
      "T1-1-3\tfastq\t../raw_reads/T1-1-3-12S_R1.fastq.gz\t../raw_reads/T1-1-3-12S_R2.fastq.gz\r\n",
      "T1-1-4\tfastq\t../raw_reads/T1-1-4-12S_R1.fastq.gz\t../raw_reads/T1-1-4-12S_R2.fastq.gz\r\n",
      "T1-1-5\tfastq\t../raw_reads/T1-1-5-12S_R1.fastq.gz\t../raw_reads/T1-1-5-12S_R2.fastq.gz\r\n",
      "T1-2-1\tfastq\t../raw_reads/T1-2-1-12S_R1.fastq.gz\t../raw_reads/T1-2-1-12S_R2.fastq.gz\r\n",
      "T1-2-2\tfastq\t../raw_reads/T1-2-2-12S_R1.fastq.gz\t../raw_reads/T1-2-2-12S_R2.fastq.gz\r\n",
      "T1-2-3\tfastq\t../raw_reads/T1-2-3-12S_R1.fastq.gz\t../raw_reads/T1-2-3-12S_R2.fastq.gz\r\n",
      "T1-2-4\tfastq\t../raw_reads/T1-2-4-12S_R1.fastq.gz\t../raw_reads/T1-2-4-12S_R2.fastq.gz\r\n",
      "T1-2-5\tfastq\t../raw_reads/T1-2-5-12S_R1.fastq.gz\t../raw_reads/T1-2-5-12S_R2.fastq.gz\r\n",
      "T1-3-1\tfastq\t../raw_reads/T1-3-1-12S_R1.fastq.gz\t../raw_reads/T1-3-1-12S_R2.fastq.gz\r\n",
      "T1-3-2\tfastq\t../raw_reads/T1-3-2-12S_R1.fastq.gz\t../raw_reads/T1-3-2-12S_R2.fastq.gz\r\n",
      "T1-3-3\tfastq\t../raw_reads/T1-3-3-12S_R1.fastq.gz\t../raw_reads/T1-3-3-12S_R2.fastq.gz\r\n",
      "T1-3-4\tfastq\t../raw_reads/T1-3-4-12S_R1.fastq.gz\t../raw_reads/T1-3-4-12S_R2.fastq.gz\r\n",
      "T1-3-5\tfastq\t../raw_reads/T1-3-5-12S_R1.fastq.gz\t../raw_reads/T1-3-5-12S_R2.fastq.gz\r\n",
      "T1-4-1\tfastq\t../raw_reads/T1-4-1-12S_R1.fastq.gz\t../raw_reads/T1-4-1-12S_R2.fastq.gz\r\n",
      "T1-4-2\tfastq\t../raw_reads/T1-4-2-12S_R1.fastq.gz\t../raw_reads/T1-4-2-12S_R2.fastq.gz\r\n",
      "T1-4-3\tfastq\t../raw_reads/T1-4-3-12S_R1.fastq.gz\t../raw_reads/T1-4-3-12S_R2.fastq.gz\r\n",
      "T1-4-4\tfastq\t../raw_reads/T1-4-4-12S_R1.fastq.gz\t../raw_reads/T1-4-4-12S_R2.fastq.gz\r\n",
      "T1-4-5\tfastq\t../raw_reads/T1-4-5-12S_R1.fastq.gz\t../raw_reads/T1-4-5-12S_R2.fastq.gz\r\n",
      "T2-1-1\tfastq\t../raw_reads/T2-1-1-12S_R1.fastq.gz\t../raw_reads/T2-1-1-12S_R2.fastq.gz\r\n",
      "T2-1-2\tfastq\t../raw_reads/T2-1-2-12S_R1.fastq.gz\t../raw_reads/T2-1-2-12S_R2.fastq.gz\r\n",
      "T2-1-3\tfastq\t../raw_reads/T2-1-3-12S_R1.fastq.gz\t../raw_reads/T2-1-3-12S_R2.fastq.gz\r\n",
      "T2-1-4\tfastq\t../raw_reads/T2-1-4-12S_R1.fastq.gz\t../raw_reads/T2-1-4-12S_R2.fastq.gz\r\n",
      "T2-1-5\tfastq\t../raw_reads/T2-1-5-12S_R1.fastq.gz\t../raw_reads/T2-1-5-12S_R2.fastq.gz\r\n",
      "T2-2-1\tfastq\t../raw_reads/T2-2-1-12S_R1.fastq.gz\t../raw_reads/T2-2-1-12S_R2.fastq.gz\r\n",
      "T2-2-2\tfastq\t../raw_reads/T2-2-2-12S_R1.fastq.gz\t../raw_reads/T2-2-2-12S_R2.fastq.gz\r\n",
      "T2-2-3\tfastq\t../raw_reads/T2-2-3-12S_R1.fastq.gz\t../raw_reads/T2-2-3-12S_R2.fastq.gz\r\n",
      "T2-2-4\tfastq\t../raw_reads/T2-2-4-12S_R1.fastq.gz\t../raw_reads/T2-2-4-12S_R2.fastq.gz\r\n",
      "T2-2-5\tfastq\t../raw_reads/T2-2-5-12S_R1.fastq.gz\t../raw_reads/T2-2-5-12S_R2.fastq.gz\r\n",
      "T2-3-1\tfastq\t../raw_reads/T2-3-1-12S_R1.fastq.gz\t../raw_reads/T2-3-1-12S_R2.fastq.gz\r\n",
      "T2-3-2\tfastq\t../raw_reads/T2-3-2-12S_R1.fastq.gz\t../raw_reads/T2-3-2-12S_R2.fastq.gz\r\n",
      "T2-3-3\tfastq\t../raw_reads/T2-3-3-12S_R1.fastq.gz\t../raw_reads/T2-3-3-12S_R2.fastq.gz\r\n",
      "T2-3-4\tfastq\t../raw_reads/T2-3-4-12S_R1.fastq.gz\t../raw_reads/T2-3-4-12S_R2.fastq.gz\r\n",
      "T2-3-5\tfastq\t../raw_reads/T2-3-5-12S_R1.fastq.gz\t../raw_reads/T2-3-5-12S_R2.fastq.gz\r\n",
      "T2-4-1\tfastq\t../raw_reads/T2-4-1-12S_R1.fastq.gz\t../raw_reads/T2-4-1-12S_R2.fastq.gz\r\n",
      "T2-4-2\tfastq\t../raw_reads/T2-4-2-12S_R1.fastq.gz\t../raw_reads/T2-4-2-12S_R2.fastq.gz\r\n",
      "T2-4-3\tfastq\t../raw_reads/T2-4-3-12S_R1.fastq.gz\t../raw_reads/T2-4-3-12S_R2.fastq.gz\r\n",
      "T2-4-4\tfastq\t../raw_reads/T2-4-4-12S_R1.fastq.gz\t../raw_reads/T2-4-4-12S_R2.fastq.gz\r\n",
      "T2-4-5\tfastq\t../raw_reads/T2-4-5-12S_R1.fastq.gz\t../raw_reads/T2-4-5-12S_R2.fastq.gz\r\n",
      "T3-1-1\tfastq\t../raw_reads/T3-1-1-12S_R1.fastq.gz\t../raw_reads/T3-1-1-12S_R2.fastq.gz\r\n",
      "T3-1-2\tfastq\t../raw_reads/T3-1-2-12S_R1.fastq.gz\t../raw_reads/T3-1-2-12S_R2.fastq.gz\r\n",
      "T3-1-3\tfastq\t../raw_reads/T3-1-3-12S_R1.fastq.gz\t../raw_reads/T3-1-3-12S_R2.fastq.gz\r\n",
      "T3-1-4\tfastq\t../raw_reads/T3-1-4-12S_R1.fastq.gz\t../raw_reads/T3-1-4-12S_R2.fastq.gz\r\n",
      "T3-1-5\tfastq\t../raw_reads/T3-1-5-12S_R1.fastq.gz\t../raw_reads/T3-1-5-12S_R2.fastq.gz\r\n",
      "T3-2-1\tfastq\t../raw_reads/T3-2-1-12S_R1.fastq.gz\t../raw_reads/T3-2-1-12S_R2.fastq.gz\r\n",
      "T3-2-2\tfastq\t../raw_reads/T3-2-2-12S_R1.fastq.gz\t../raw_reads/T3-2-2-12S_R2.fastq.gz\r\n",
      "T3-2-3\tfastq\t../raw_reads/T3-2-3-12S_R1.fastq.gz\t../raw_reads/T3-2-3-12S_R2.fastq.gz\r\n",
      "T3-2-4\tfastq\t../raw_reads/T3-2-4-12S_R1.fastq.gz\t../raw_reads/T3-2-4-12S_R2.fastq.gz\r\n",
      "T3-2-5\tfastq\t../raw_reads/T3-2-5-12S_R1.fastq.gz\t../raw_reads/T3-2-5-12S_R2.fastq.gz\r\n",
      "T3-3-1\tfastq\t../raw_reads/T3-3-1-12S_R1.fastq.gz\t../raw_reads/T3-3-1-12S_R2.fastq.gz\r\n",
      "T3-3-2\tfastq\t../raw_reads/T3-3-2-12S_R1.fastq.gz\t../raw_reads/T3-3-2-12S_R2.fastq.gz\r\n",
      "T3-3-3\tfastq\t../raw_reads/T3-3-3-12S_R1.fastq.gz\t../raw_reads/T3-3-3-12S_R2.fastq.gz\r\n",
      "T3-3-4\tfastq\t../raw_reads/T3-3-4-12S_R1.fastq.gz\t../raw_reads/T3-3-4-12S_R2.fastq.gz\r\n",
      "T3-3-5\tfastq\t../raw_reads/T3-3-5-12S_R1.fastq.gz\t../raw_reads/T3-3-5-12S_R2.fastq.gz\r\n",
      "T3-4-1\tfastq\t../raw_reads/T3-4-1-12S_R1.fastq.gz\t../raw_reads/T3-4-1-12S_R2.fastq.gz\r\n",
      "T3-4-2\tfastq\t../raw_reads/T3-4-2-12S_R1.fastq.gz\t../raw_reads/T3-4-2-12S_R2.fastq.gz\r\n",
      "T3-4-3\tfastq\t../raw_reads/T3-4-3-12S_R1.fastq.gz\t../raw_reads/T3-4-3-12S_R2.fastq.gz\r\n",
      "T3-4-4\tfastq\t../raw_reads/T3-4-4-12S_R1.fastq.gz\t../raw_reads/T3-4-4-12S_R2.fastq.gz\r\n",
      "T3-4-5\tfastq\t../raw_reads/T3-4-5-12S_R1.fastq.gz\t../raw_reads/T3-4-5-12S_R2.fastq.gz\r\n",
      "T4-1-1\tfastq\t../raw_reads/T4-1-1-12S_R1.fastq.gz\t../raw_reads/T4-1-1-12S_R2.fastq.gz\r\n",
      "T4-1-2\tfastq\t../raw_reads/T4-1-2-12S_R1.fastq.gz\t../raw_reads/T4-1-2-12S_R2.fastq.gz\r\n",
      "T4-1-3\tfastq\t../raw_reads/T4-1-3-12S_R1.fastq.gz\t../raw_reads/T4-1-3-12S_R2.fastq.gz\r\n",
      "T4-1-4\tfastq\t../raw_reads/T4-1-4-12S_R1.fastq.gz\t../raw_reads/T4-1-4-12S_R2.fastq.gz\r\n",
      "T4-1-5\tfastq\t../raw_reads/T4-1-5-12S_R1.fastq.gz\t../raw_reads/T4-1-5-12S_R2.fastq.gz\r\n",
      "T4-2-1\tfastq\t../raw_reads/T4-2-1-12S_R1.fastq.gz\t../raw_reads/T4-2-1-12S_R2.fastq.gz\r\n",
      "T4-2-2\tfastq\t../raw_reads/T4-2-2-12S_R1.fastq.gz\t../raw_reads/T4-2-2-12S_R2.fastq.gz\r\n",
      "T4-2-3\tfastq\t../raw_reads/T4-2-3-12S_R1.fastq.gz\t../raw_reads/T4-2-3-12S_R2.fastq.gz\r\n",
      "T4-2-4\tfastq\t../raw_reads/T4-2-4-12S_R1.fastq.gz\t../raw_reads/T4-2-4-12S_R2.fastq.gz\r\n",
      "T4-2-5\tfastq\t../raw_reads/T4-2-5-12S_R1.fastq.gz\t../raw_reads/T4-2-5-12S_R2.fastq.gz\r\n",
      "T4-3-1\tfastq\t../raw_reads/T4-3-1-12S_R1.fastq.gz\t../raw_reads/T4-3-1-12S_R2.fastq.gz\r\n",
      "T4-3-2\tfastq\t../raw_reads/T4-3-2-12S_R1.fastq.gz\t../raw_reads/T4-3-2-12S_R2.fastq.gz\r\n",
      "T4-3-3\tfastq\t../raw_reads/T4-3-3-12S_R1.fastq.gz\t../raw_reads/T4-3-3-12S_R2.fastq.gz\r\n",
      "T4-3-4\tfastq\t../raw_reads/T4-3-4-12S_R1.fastq.gz\t../raw_reads/T4-3-4-12S_R2.fastq.gz\r\n",
      "T4-3-5\tfastq\t../raw_reads/T4-3-5-12S_R1.fastq.gz\t../raw_reads/T4-3-5-12S_R2.fastq.gz\r\n",
      "T4-4-1\tfastq\t../raw_reads/T4-4-1-12S_R1.fastq.gz\t../raw_reads/T4-4-1-12S_R2.fastq.gz\r\n",
      "T4-4-2\tfastq\t../raw_reads/T4-4-2-12S_R1.fastq.gz\t../raw_reads/T4-4-2-12S_R2.fastq.gz\r\n",
      "T4-4-3\tfastq\t../raw_reads/T4-4-3-12S_R1.fastq.gz\t../raw_reads/T4-4-3-12S_R2.fastq.gz\r\n",
      "T4-4-4\tfastq\t../raw_reads/T4-4-4-12S_R1.fastq.gz\t../raw_reads/T4-4-4-12S_R2.fastq.gz\r\n",
      "T4-4-5\tfastq\t../raw_reads/T4-4-5-12S_R1.fastq.gz\t../raw_reads/T4-4-5-12S_R2.fastq.gz\r\n",
      "T5-1-1\tfastq\t../raw_reads/T5-1-1-12S_R1.fastq.gz\t../raw_reads/T5-1-1-12S_R2.fastq.gz\r\n",
      "T5-1-2\tfastq\t../raw_reads/T5-1-2-12S_R1.fastq.gz\t../raw_reads/T5-1-2-12S_R2.fastq.gz\r\n",
      "T5-1-3\tfastq\t../raw_reads/T5-1-3-12S_R1.fastq.gz\t../raw_reads/T5-1-3-12S_R2.fastq.gz\r\n",
      "T5-1-4\tfastq\t../raw_reads/T5-1-4-12S_R1.fastq.gz\t../raw_reads/T5-1-4-12S_R2.fastq.gz\r\n",
      "T5-1-5\tfastq\t../raw_reads/T5-1-5-12S_R1.fastq.gz\t../raw_reads/T5-1-5-12S_R2.fastq.gz\r\n",
      "T5-2-1\tfastq\t../raw_reads/T5-2-1-12S_R1.fastq.gz\t../raw_reads/T5-2-1-12S_R2.fastq.gz\r\n",
      "T5-2-2\tfastq\t../raw_reads/T5-2-2-12S_R1.fastq.gz\t../raw_reads/T5-2-2-12S_R2.fastq.gz\r\n",
      "T5-2-3\tfastq\t../raw_reads/T5-2-3-12S_R1.fastq.gz\t../raw_reads/T5-2-3-12S_R2.fastq.gz\r\n",
      "T5-2-4\tfastq\t../raw_reads/T5-2-4-12S_R1.fastq.gz\t../raw_reads/T5-2-4-12S_R2.fastq.gz\r\n",
      "T5-2-5\tfastq\t../raw_reads/T5-2-5-12S_R1.fastq.gz\t../raw_reads/T5-2-5-12S_R2.fastq.gz\r\n",
      "T5-3-1\tfastq\t../raw_reads/T5-3-1-12S_R1.fastq.gz\t../raw_reads/T5-3-1-12S_R2.fastq.gz\r\n",
      "T5-3-2\tfastq\t../raw_reads/T5-3-2-12S_R1.fastq.gz\t../raw_reads/T5-3-2-12S_R2.fastq.gz\r\n",
      "T5-3-3\tfastq\t../raw_reads/T5-3-3-12S_R1.fastq.gz\t../raw_reads/T5-3-3-12S_R2.fastq.gz\r\n",
      "T5-3-4\tfastq\t../raw_reads/T5-3-4-12S_R1.fastq.gz\t../raw_reads/T5-3-4-12S_R2.fastq.gz\r\n",
      "T5-3-5\tfastq\t../raw_reads/T5-3-5-12S_R1.fastq.gz\t../raw_reads/T5-3-5-12S_R2.fastq.gz\r\n",
      "T5-4-1\tfastq\t../raw_reads/T5-4-1-12S_R1.fastq.gz\t../raw_reads/T5-4-1-12S_R2.fastq.gz\r\n",
      "T5-4-2\tfastq\t../raw_reads/T5-4-2-12S_R1.fastq.gz\t../raw_reads/T5-4-2-12S_R2.fastq.gz\r\n",
      "T5-4-3\tfastq\t../raw_reads/T5-4-3-12S_R1.fastq.gz\t../raw_reads/T5-4-3-12S_R2.fastq.gz\r\n",
      "T5-4-4\tfastq\t../raw_reads/T5-4-4-12S_R1.fastq.gz\t../raw_reads/T5-4-4-12S_R2.fastq.gz\r\n",
      "T5-4-5\tfastq\t../raw_reads/T5-4-5-12S_R1.fastq.gz\t../raw_reads/T5-4-5-12S_R2.fastq.gz\r\n",
      "TP5-1-1\tfastq\t../raw_reads/TP5-1-1-12S_R1.fastq.gz\t../raw_reads/TP5-1-1-12S_R2.fastq.gz\r\n",
      "TP5-1-2\tfastq\t../raw_reads/TP5-1-2-12S_R1.fastq.gz\t../raw_reads/TP5-1-2-12S_R2.fastq.gz\r\n",
      "TP5-1-3\tfastq\t../raw_reads/TP5-1-3-12S_R1.fastq.gz\t../raw_reads/TP5-1-3-12S_R2.fastq.gz\r\n",
      "TP5-1-4\tfastq\t../raw_reads/TP5-1-4-12S_R1.fastq.gz\t../raw_reads/TP5-1-4-12S_R2.fastq.gz\r\n",
      "TP5-1-5\tfastq\t../raw_reads/TP5-1-5-12S_R1.fastq.gz\t../raw_reads/TP5-1-5-12S_R2.fastq.gz\r\n",
      "TP5-2-1\tfastq\t../raw_reads/TP5-2-1-12S_R1.fastq.gz\t../raw_reads/TP5-2-1-12S_R2.fastq.gz\r\n",
      "TP5-2-2\tfastq\t../raw_reads/TP5-2-2-12S_R1.fastq.gz\t../raw_reads/TP5-2-2-12S_R2.fastq.gz\r\n",
      "TP5-2-3\tfastq\t../raw_reads/TP5-2-3-12S_R1.fastq.gz\t../raw_reads/TP5-2-3-12S_R2.fastq.gz\r\n",
      "TP5-2-4\tfastq\t../raw_reads/TP5-2-4-12S_R1.fastq.gz\t../raw_reads/TP5-2-4-12S_R2.fastq.gz\r\n",
      "TP5-2-5\tfastq\t../raw_reads/TP5-2-5-12S_R1.fastq.gz\t../raw_reads/TP5-2-5-12S_R2.fastq.gz\r\n",
      "TP5-3-1\tfastq\t../raw_reads/TP5-3-1-12S_R1.fastq.gz\t../raw_reads/TP5-3-1-12S_R2.fastq.gz\r\n",
      "TP5-3-2\tfastq\t../raw_reads/TP5-3-2-12S_R1.fastq.gz\t../raw_reads/TP5-3-2-12S_R2.fastq.gz\r\n",
      "TP5-3-3\tfastq\t../raw_reads/TP5-3-3-12S_R1.fastq.gz\t../raw_reads/TP5-3-3-12S_R2.fastq.gz\r\n",
      "TP5-3-4\tfastq\t../raw_reads/TP5-3-4-12S_R1.fastq.gz\t../raw_reads/TP5-3-4-12S_R2.fastq.gz\r\n",
      "TP5-3-5\tfastq\t../raw_reads/TP5-3-5-12S_R1.fastq.gz\t../raw_reads/TP5-3-5-12S_R2.fastq.gz\r\n",
      "TP5-4-1\tfastq\t../raw_reads/TP5-4-1-12S_R1.fastq.gz\t../raw_reads/TP5-4-1-12S_R2.fastq.gz\r\n",
      "TP5-4-2\tfastq\t../raw_reads/TP5-4-2-12S_R1.fastq.gz\t../raw_reads/TP5-4-2-12S_R2.fastq.gz\r\n",
      "TP5-4-3\tfastq\t../raw_reads/TP5-4-3-12S_R1.fastq.gz\t../raw_reads/TP5-4-3-12S_R2.fastq.gz\r\n",
      "TP5-4-4\tfastq\t../raw_reads/TP5-4-4-12S_R1.fastq.gz\t../raw_reads/TP5-4-4-12S_R2.fastq.gz\r\n",
      "TP5-4-5\tfastq\t../raw_reads/TP5-4-5-12S_R1.fastq.gz\t../raw_reads/TP5-4-5-12S_R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!cat Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "The 12S amplicon sequenced here is only 106bp long. Readlength used in the MiSeq run was 2x250bp. Our reads are thus longer than our amplicon and we so expect to find primer/adapter sequences in our reads that need to be removed as part of the raw data processing. \n",
    "\n",
    "Specifically, forward reads are expected to contain the reverse complement of the reverse primer plus the reverse Illumina adapter (FA501-FA508; FB501-FB508), and reverse reads will contain reverse complements of the forward primers and adapters (RA701-RA712;RB701-RB712).\n",
    "\n",
    "The expected sequences have been produced from `12S-adapters.fasta` (see [here](https://github.com/HullUni-bioinformatics/Li_et_al_2017/blob/master/12S-adapters.fasta)) into reverse complements `12S-adapters_rc.fasta` and will be used in the trimming algorithm."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FA501_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACGTACGATGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FA502_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACAGATAGTGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FA503_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACTCGCTAGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FA504_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACACGCAGGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FA505_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACTCGATGAGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FA506_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACACTCACGGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FA507_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAAGATATCCGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FA508_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACGGTGTCGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB501_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATATATAGTAGGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB502_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATATAGTAACGGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB503_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAGTGACTCTGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB504_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAGTCTCGTAGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB505_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACGAGACGTGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB506_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACTCGTCGAGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB507_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACACGATCGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "FB508_rc\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATATATCTGACGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
      "RA701_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGAGAGTTATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA702_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGACATAGTATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA703_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACGCTACTATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA704_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACTCACTGATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA705_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTGAGTACGATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA706_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCTGCGTAGATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA707_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTAGTCTCCATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA708_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGAGCGACATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA709_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACTACGACATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA710_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGTCTGCTAATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA711_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGTCTATGAATCTCGTATGCCGTCTTCTGCTTG\n",
      "RA712_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTATAGCGAATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB701_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCTCGACTTATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB702_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGAAGTATATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB703_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTAGCAGCTATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB704_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTCTCTATGATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB705_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGATCTACGATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB706_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGTAACGAGATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB707_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACGTGCGCATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB708_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTATAGTACCATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB709_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGCGTATACATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB710_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTGCTCGTAATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB711_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTAACGCTGAATCTCGTATGCCGTCTTCTGCTTG\n",
      "RB712_rc\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGTAGCGAATCTCGTATGCCGTCTTCTGCTTG\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "40"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "for record in SeqIO.parse(\"../12S-adapters.fasta\", \"fasta\"):\n",
    "    print record.id + \"_rc\"\n",
    "    print record.seq.reverse_complement()\n",
    "\n",
    "def make_rc_record(record):\n",
    "    \"\"\"Returns a new SeqRecord with the reverse complement sequence.\"\"\"\n",
    "    return SeqRecord(seq = record.seq.reverse_complement(), \\\n",
    "                 id = record.id + \"_rc\", \\\n",
    "                 description = \"reverse complement\")\n",
    "\n",
    "records = map(make_rc_record, SeqIO.parse(\"../12S-adapters.fasta\", \"fasta\"))\n",
    "SeqIO.write(records, \"../12S-adapters_rc.fasta\", \"fasta\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">FA501_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACGTACGATGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FA502_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACAGATAGTGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FA503_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACTCGCTAGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FA504_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACACGCAGGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FA505_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACTCGATGAGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FA506_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACACTCACGGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FA507_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAAGATATCCGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FA508_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACGGTGTCGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB501_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATATATAGTAGGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB502_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATATAGTAACGGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB503_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAGTGACTCTGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB504_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAGTCTCGTAGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB505_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACGAGACGTGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB506_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATACTCGTCGAGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB507_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATAACACGATCGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">FB508_rc reverse complement\r\n",
      "GGGGTATCTAATCCCAGTCCAATTACCATATATCTGACGTGTAGATCTCGGTGGTCGCCG\r\n",
      "TATCATT\r\n",
      ">RA701_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGAGAGTTATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA702_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGACATAGTATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA703_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACGCTACTATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA704_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACTCACTGATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA705_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTGAGTACGATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA706_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCTGCGTAGATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA707_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTAGTCTCCATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA708_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGAGCGACATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA709_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACTACGACATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA710_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGTCTGCTAATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA711_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGTCTATGAATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RA712_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTATAGCGAATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB701_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCTCGACTTATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB702_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGAAGTATATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB703_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTAGCAGCTATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB704_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTCTCTATGATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB705_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGATCTACGATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB706_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGTAACGAGATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB707_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTACGTGCGCATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB708_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTATAGTACCATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB709_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTGCGTATACATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB710_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTTGCTCGTAATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB711_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTAACGCTGAATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n",
      ">RB712_rc reverse complement\r\n",
      "CTAGAGGAGCCTGTTCTAGGCTGACTGACTCGTAGCGAATCTCGTATGCCGTCTTCTGCT\r\n",
      "TG\r\n"
     ]
    }
   ],
   "source": [
    "!cat ../12S-adapters_rc.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Raw data trimming, removal of adapter sequences and merging of reads using the `metaBEAT` pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: metaBEAT.py [-h] [-Q <FILE>] [-B <FILE>] [--g_queries <FILE>] [-v] [-s]\r\n",
      "                   [-f] [-p] [-k] [-t] [-b] [-m <string>] [-n <INT>] [-E] [-e]\r\n",
      "                   [--read_stats_off] [--PCR_primer <FILE>]\r\n",
      "                   [--trim_adapter <FILE>] [--trim_qual <INT>] [--phred <INT>]\r\n",
      "                   [--trim_window <INT>] [--read_crop <INT>]\r\n",
      "                   [--trim_minlength <INT>] [--merge] [--product_length <INT>]\r\n",
      "                   [--merged_only] [--forward_only] [--length_filter <INT>]\r\n",
      "                   [--length_deviation <FLOAT>] [-R <FILE>] [--gb_out <FILE>]\r\n",
      "                   [--rec_check] [--gb_to_taxid <FILE>] [--cluster]\r\n",
      "                   [--clust_match <FLOAT>] [--clust_cov <INT>]\r\n",
      "                   [--blast_db <PATH>] [--blast_xml <PATH>]\r\n",
      "                   [--min_ident <FLOAT>] [--min_ali_length <FLOAT>]\r\n",
      "                   [--bitscore_skim_LCA <FLOAT>] [--bitscore_skim_adjust_off]\r\n",
      "                   [--min_bit <INT>] [--refpkg <DIR>] [--jplace <FILE>]\r\n",
      "                   [--jellyfish_hash_size <STR>] [--kraken_db <DIR>]\r\n",
      "                   [--kraken_score_threshold <FLOAT>] [--rm_kraken_db]\r\n",
      "                   [-o OUTPUT_PREFIX] [--metadata METADATA]\r\n",
      "                   [-@ <email-address>] [--version]\r\n",
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool version:\r\n",
      "v.0.97.8-global\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -Q <FILE>, --querylist <FILE>\r\n",
      "                        file containing a list of query files\r\n",
      "  -B <FILE>, --BIOM_input <FILE>\r\n",
      "                        OTU table in BIOM format\r\n",
      "  --g_queries <FILE>    fasta file containing query sequences (in combination\r\n",
      "                        with '-B' sequence headers are expected to match)\r\n",
      "  -v, --verbose         turn verbose output on\r\n",
      "  -s, --seqinfo         write out seq_info.csv file\r\n",
      "  -f, --fasta           write out ref.fasta file\r\n",
      "  -p, --pplace          perform phylogenetic placement\r\n",
      "  -k, --kraken          perform phylogenetic placement\r\n",
      "  -t, --taxids          write out taxid.txt file\r\n",
      "  -b, --blast           compile local blast db and blast queries\r\n",
      "  -m <string>, --marker <string>\r\n",
      "                        marker ID (default: marker)\r\n",
      "  -n <INT>, --n_threads <INT>\r\n",
      "                        Number of threads (default: 1)\r\n",
      "  -E, --extract_centroid_reads\r\n",
      "                        extract centroid reads to files\r\n",
      "  -e, --extract_all_reads\r\n",
      "                        extract reads to files\r\n",
      "  --read_stats_off      ommit writing read stats to file\r\n",
      "  --version             show program's version number and exit\r\n",
      "\r\n",
      "Query preprocessing:\r\n",
      "  The parameters in this group affect how the query sequences are processed\r\n",
      "\r\n",
      "  --PCR_primer <FILE>   PCR primers (provided in fasta file) to be clipped\r\n",
      "                        from reads\r\n",
      "  --trim_adapter <FILE>\r\n",
      "                        trim adapters provided in file\r\n",
      "  --trim_qual <INT>     minimum phred quality score (default: 30)\r\n",
      "  --phred <INT>         phred quality score offset - 33 or 64 (default: 33)\r\n",
      "  --trim_window <INT>   sliding window size (default: 5) for trimming; if\r\n",
      "                        average quality drops below the specified minimum\r\n",
      "                        quality all subsequent bases are removed from the\r\n",
      "                        reads\r\n",
      "  --read_crop <INT>     Crop reads to this length if they are longer than that\r\n",
      "                        (default: off)\r\n",
      "  --trim_minlength <INT>\r\n",
      "                        minimum length of reads to be retained after trimming\r\n",
      "                        (default: 50)\r\n",
      "  --merge               attempt to merge paired-end reads\r\n",
      "  --product_length <INT>\r\n",
      "                        estimated length of PCR product (specifying this\r\n",
      "                        option increases merging efficiency)\r\n",
      "  --merged_only         only process successfully merged read-pairs\r\n",
      "  --forward_only        only process sequences that contain forward reads\r\n",
      "                        (i.e. unmerged forward reads and merged reads)\r\n",
      "  --length_filter <INT>\r\n",
      "                        only process reads, which are within +/- 10 percent of\r\n",
      "                        this length\r\n",
      "  --length_deviation <FLOAT>\r\n",
      "                        allowed deviation (in percent) from length specified\r\n",
      "                        by --length_filter (default=0.1)\r\n",
      "\r\n",
      "Reference:\r\n",
      "  The parameters in this group affect the reference to be used in the\r\n",
      "  analyses\r\n",
      "\r\n",
      "  -R <FILE>, --REFlist <FILE>\r\n",
      "                        file containing a list of files to be used as\r\n",
      "                        reference sequences\r\n",
      "  --gb_out <FILE>       output the corrected gb file\r\n",
      "  --rec_check           check records to be used as reference\r\n",
      "  --gb_to_taxid <FILE>  comma delimited file containing 'gb accession,taxid'\r\n",
      "                        for a list of taxa\r\n",
      "\r\n",
      "Query clustering options:\r\n",
      "  The parameters in this group affect read clustering\r\n",
      "\r\n",
      "  --cluster             perform clustering of query sequences using vsearch\r\n",
      "  --clust_match <FLOAT>\r\n",
      "                        identity threshold for clustering in percent (default:\r\n",
      "                        1)\r\n",
      "  --clust_cov <INT>     minimum number of records in cluster (default: 1)\r\n",
      "\r\n",
      "BLAST search:\r\n",
      "  The parameters in this group affect BLAST search and BLAST based taxonomic\r\n",
      "  assignment\r\n",
      "\r\n",
      "  --blast_db <PATH>     path to precompiled blast database\r\n",
      "  --blast_xml <PATH>    path to Blast result in xml format\r\n",
      "  --min_ident <FLOAT>   minimum identity threshold in percent (default: 0.80)\r\n",
      "  --min_ali_length <FLOAT>\r\n",
      "                        minimum alignment length in percent of total query\r\n",
      "                        length (default: 0.95)\r\n",
      "  --bitscore_skim_LCA <FLOAT>\r\n",
      "                        Only BLAST hits with bitscores differing by less than\r\n",
      "                        this factor from the top hit (bitscore skim window)\r\n",
      "                        will be considered for LCA (0-1; default: 0.1)\r\n",
      "  --bitscore_skim_adjust_off\r\n",
      "                        Per default a 100% identity BLAST top hit across the\r\n",
      "                        minimum alignment length triggers an adjustment of the\r\n",
      "                        bitscore skim window to '0', i.e. only hits with\r\n",
      "                        bitscores as good as the top hit are considered for\r\n",
      "                        LCA. This flag switches this behaviour off.\r\n",
      "  --min_bit <INT>       minimum bitscore (default: 80)\r\n",
      "\r\n",
      "Phylogenetic placement:\r\n",
      "  The parameters in this group affect phylogenetic placement\r\n",
      "\r\n",
      "  --refpkg <DIR>        PATH to refpkg for pplacer\r\n",
      "  --jplace <FILE>       phylogenetic placement result from prefious pplacer\r\n",
      "                        run in *.jplace format\r\n",
      "\r\n",
      "Kraken:\r\n",
      "  The parameters in this group affect taxonomic assignment using Kraken\r\n",
      "\r\n",
      "  --jellyfish_hash_size <STR>\r\n",
      "                        jellyfish hash size to control memory usage during\r\n",
      "                        kraken database building. A table size of '6400M' will\r\n",
      "                        require ~44G of RAM. '2700M' -> 20G RAM.\r\n",
      "  --kraken_db <DIR>     PATH to a Kraken database\r\n",
      "  --kraken_score_threshold <FLOAT>\r\n",
      "                        minimum proportion of k-mers to support assignment\r\n",
      "                        (0-1; default: 0)\r\n",
      "  --rm_kraken_db        Remove Kraken database after successful completion\r\n",
      "\r\n",
      "BIOM OUTPUT:\r\n",
      "  The arguments in this groups affect the output in BIOM format\r\n",
      "\r\n",
      "  -o OUTPUT_PREFIX, --output_prefix OUTPUT_PREFIX\r\n",
      "                        prefix for BIOM output files (default='metaBEAT')\r\n",
      "  --metadata METADATA   comma delimited file containing metadata (optional)\r\n",
      "\r\n",
      "Entrez identification:\r\n",
      "  metaBEAT is querying the NCBI Entrez databases, please provide an email\r\n",
      "  address for identification\r\n",
      "\r\n",
      "  -@ <email-address>, --email <email-address>\r\n",
      "                        provide your email address for identification to NCBI\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-Q Querymap.txt \\\n",
    "--trim_qual 30 \\\n",
    "--trim_adapter ../12S-adapters_rc.fasta \\\n",
    "--trim_minlength 90 \\\n",
    "--merge \\\n",
    "--product_length 110 \\\n",
    "--forward_only \\\n",
    "--read_crop 106 \\\n",
    "-@ haikuilee@gmail.com \\\n",
    "-n 5 -v &> log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Query survival after trimming__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some stats on the read counts before/after trimming, merging etc. are summarized for you in `metaBEAT_read_stats.csv`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes.AxesSubplot at 0x7f8a8c4ef210>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXIAAAEACAYAAACuzv3DAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAGMpJREFUeJzt3XuMZGlZx/Hv9ja7MMsuG1jkuqS5yEUkbEAIQdBHRUSu\nBjUKBpA/iAkqEIPCGoH5S0OCuibeoggrBNCAAZeL3HRerioiM+wuLAQWRq4iQoZdGYiM3f5xqnZq\nup/Tdar6nPOrevv3SXq76tSpU9+prX66+u2ubjAzMzMzMzMzMzMzMzMzMzMzMzMzG8zzgeuA6yen\nzcxsjfwgzRC/NXA+8B7g3tIiMzMDYKPjfvcH/hX4LvB/wPuApw4VZWZm3XUd5NcDjwZuDxwBngDc\nfagoMzPrbrPjfp8CXg68G/g2cBzYHirKzMy6O2/J6/0u8AXgz2e2fRm464GLzMwOjxuB+4x5g983\neX8P4Abgkl2X74wZ09FRdUDiqDogcVQdkDiqDmhxdIBj7sBOy1unj6sRmzp/nA/RdFBH1QGJXuZm\n16UVgDcBdwC+BzwXuKmPgIFtqQMSW+qAxJY6ILGlDmixpQ5IbKkDElvqgMSWOmAoiwzyHxmswszM\nVsIqLq2EOiAR6oBEqAMSoQ5oEQMc86BLKyM2df44H6LpoEIdkOhlbi77zc7MTs/HMzssdto/ns+7\n5T8ja2mS9dSql7nZ9efI11WoAxKhDkiEOiAR6oAWoQ5IhDogEeqARKgDhlL7IDczq56XVsz0vLRy\neHlpxczM6h/koQ5IhDogEeqARKgDWoQ6IBHqgESoAxKhDhhK7YPczKx6XiM30/Ma+eHlNXIzM6t/\nkIc6IBHqgESoAxKhDmgR6oBEqAMSoQ5IhDpgKLUPcjOz6nmN3EzPa+SHl9fIzcys/kEe6oBEqAMS\noQ5IhDqgRagDEqEOSIQ6IBHqgKHUPsjNzGzGlcAngOuA1wMX7rp8FX8fudk6OOjvIx+xyR/nPRv1\n/twCPsfZ4f23wLOUQWYV8SA/vHq5P7surdxE87c6j9D8ebgjwJf7CBhYqAMSoQ5IhDogEeqAFqEO\nSIQ6IBHqgESoA4bSdZB/E/h94AvAV4BTwHuHijIzs+66/vHlewMvoFli+RbwRuCXgNft2u9q4OTk\n9CngBFAm52PyfuzzzLnc55u3VephZtuq9Ax8vm23W+y6YOPbsH1k906weTOcuaSHPvKHRVtP2/+/\nzWvgzMV7OzdOw/YTDtC37HnmXD7G7QfNLB3dLwCvnDn/DOBPdu3jtTOz5SyzRj70GnZfx/da+/5G\nXSP/FPAI4DY0r0J6DPDJPgIGFuqARKgDEqEOSIQ6oEWoA/Yq6oBMqAMSoQ4YStdB/nHgNcBHgWsn\n2/5ikCIzM1uIf9eKmd4yv2tl6N+F0tfx/Ttb9ufftWJmZvUP8lAHJEIdkAh1QCLUAS1CHbBXUQdk\nQh2QCHXAUGof5GZm1fMauZme18gPL6+Rm5lZ/YM81AGJUAckQh2QCHVAi1AH7FXUAZlQByRCHTCU\n2ge5mVn1vEZupuc18sPLa+RmZlb/IA91QCLUAYlQByRCHdAi1AF7FXVAJtQBiVAHDKX2QW5mVj2v\nkZvpeY388PIauZmZ1T/IQx2QCHVAItQBiVAHtAh1wF5FHZAJdUAi1AFDqX2Qm5lVz2vkZnpeIz+8\nRl0jvx9wfObtW8DzDnrjZmamsQF8Fbh81/ZV/GOqoQ5IhDogEeqARKgDWsQAxzzgH18+top/fDl6\nOk6fYu4e4xv1jy/PegxwI/DFPgLMzGx8rwKem2xfxWfkZuvggM/IB3mm29fxV+kZ+Srq5X7YXHD/\nC4AnAS9qufxq4OTk9CngBGd/Niom733e531+z/m23TaBMy0f7GWB/TdOw/YTFuhrOf4t5l0/OMfu\nzRvAdtK5eTOcuWSB46/b+enpLYSeAryz5bJV/Awb6oBEqAMSoQ5IhDqgRQxwzDnPyOdt371GvhLP\npGP+cUZ/ph5z9xifZI38acAb+rhhMzPrxyI/v3gR8B/APYGbk8v9c+Rmy5nzc+RtP4e96HbJz3/v\ncxz/fDk9zU2/IMhMz4N8+eOvu17mZu0v0Q91QCLUAYlQByRCHdAi1AF7FXVAJtQBiVAHDKX2QW5m\nVj0vrZjpeWll+eOvOy+tmJlZ/YM81AGJUAckQh2QCHVAi1AH7FXUAZlQByRCHTCU2ge5mVn1vEZu\npuc18uWPv+68Rm5mZvUP8lAHJEIdkAh1QCLUAS1CHbBXUQdkQh2QCHXAUGof5GZm1fMauZme18iX\nP/668xq5mZnVP8hDHZAIdUAi1AGJUAe0CHXAXkUdkAl1QCLUAUOpfZCbmVXPa+Rmel4jX/74685r\n5GZmttggvxR4E3AD8EngEYMU9SvUAYlQByRCHZAIdUCLUAfsVdQBmVAHJEIdMJTNBfb9I+AdwM9N\nrnfRIEVmZraQrmsztwOOA/faZx+vkZstx2vkyx9/3Y26Rn5P4OvAq4GPAX8JHDnojZuZ2cF1XVrZ\nBB4C/Brwb8BVwIuBl+7a72rg5OT0KeAEZxfwYvJ+zPNXTFpVt5+dn25blZ7ZllXpAXgB+sfP5Pzm\nTXDmYvbYOA3bF+3dv+38xtthu+UJUNvV2i6fbotk37b9z9l5csHmNfm/bfb4s8fbBM4kT6U3gO38\nMK092fFvsfuCPs6vwjyYnt5C4M7A52fOPwp426592r42VAp1QCLUAYlQByRCHTBjB3Ymb8dmTi/8\nmJ+57s6u42Tb97tsv6a2/fto6rr92AGOM5gY8NjLGn1uvh+47+T0UeDluy5fxUFu1oe+hs5Ag/xA\nA3KE212ZQb6Kevn3LrLI/mDglcAFwI3As4Fv7Qo6TN+ksMNjJ/94G/obf/td1ts3OxdsGmP7oZoj\nKzc3V/EzaagDEqEOSIQ6IBHqgBnJksEqPSP30kpHMeCxl9XLv9ev7DQzW3P+XStm83lpZdTth2qO\n9DI3/YzczGzN1T7IQx2QCHVAItQBiVAH5Io6IFHUAYmiDsiEOmAotQ9yM7PqeY3cbD6vkY+6/VDN\nEa+Rm5lZ/YM81AGJUAckQh2QCHVArqgDEkUdkCjqgEyoA4ZS+yA3M6ue18jN5vMa+ajbD9Uc8Rq5\nmZnVP8hDHZAIdUAi1AGJUAfkijogUdQBiaIOyIQ6YCi1D3Izs+p5jdxsPq+Rj7r9UM0Rr5Gbmdli\ng/wkcC1wHPjIIDX9C3VAItQBiVAHJEIdkCvqgERRBySKOiAT6oChdP3jy9B8CRDAN4dJMTOzZSyy\nNvN54IeAb7Rc7jVyq5XXyEfdfqjmyOhr5DvAe4GPAs856A2bmdn47jJ5f0fgBPDoXZcP+bf2lhXq\ngESoAxKhDkiEOmBG8rco/Tc799/uv9nZUS//3kXWyL86ef914M3Aw4EP7NrnappvigKcohn4ZXI+\nJu/HPH+F+Paz88y53OcbV6xWz3QzM5s3gO3kA3ETOLN3857rd8lo23+6bZH9ibYL+jn+7P4nFtx/\n9/F3X7Dxbdg+snsn2LwZzlySHCg7vwrzYHp6C4EjwMWT0xcBHwIeu2ufVXxGbtaHgZ619vWM/EDP\ndEX/tj4719qoz8jvRPMsfHqd1wHv7iPAzMxWxyp+Zgx1QCLUAYlQByRCHTAjWftdpWfkXZvm/dv6\n/DccO8BxFu3sLBbYdyy9zE2/stPMbM35d62YzbeTP3Ea+meq+76Ntf458n0613ru9DI3/YzczGzN\n1T7IQx2QCHVAItQBiVAH5Io6IFHUAYmiDsiEOmAotQ9yM7PqeY3cbD6vkY+63Wvki/IzcjOzNVf7\nIA91QCLUAYlQByRCHZAr6oBEUQckijogE+qAodQ+yM3Mquc1crP5vEY+6navkS/Kz8jNzNZc7YM8\n1AGJUAckQh2QCHVArqgDEkUdkCjqgEyoA4ZS+yA3M6ue18jN5vMa+ajbvUa+KD8jNzNbc7UP8lAH\nJEIdkAh1QCLUAbmiDkgUdUCiqAMyoQ4YyiKD/HzgOPDWgVrMzGwJi6zN/AbwUJq/3fnk5HKvkVut\nvEY+6navkS+q6zPyuwOPB17Zx42amVl/ug7yPwR+E9gesGUIoQ5IhDogEeqARKgDckUdkCjqgERR\nB2RCHTCULoP8icB/0ayP+9m4mdmK2eywzyNp1sQfD9wauAR4DfDMZN+rgZOT06eAE5z91ByT92Of\nZ87lPt+8rVIPM9tWpGe/zEUO29f+023BuXdT2/4bwPY+f7F9v+Mv2rN7n0WOvwmcaencr2dt5sH0\n9BY9WvQZ9o8CLwSelFzmb3ZarSr5Zmet29d67sheELTPZ/WVE+qARKgDEqEOSIQ6IFfUAYmiDkgU\ndUAm1AFD6bK0Mut9kzczM1sR/l0rZvN5aWWlt6/13PHvWjEzs/oHeagDEqEOSIQ6IBHqgFxRBySK\nOiBR1AGZUAcMpfZBbmZWPa+Rm83nNfKV3r7Wc8dr5GZmVv8gD3VAItQBiVAHJEIdkCvqgERRBySK\nOiAT6oCh1D7Izcyq5zVys/m8Rr7S29d67niN3MzM6h/koQ5IhDogEeqARKgDckUdkCjqgERRB2RC\nHTCU2ge5mVn1vEZuNp/XyFd6+1rPHa+Rm5lZ/YM81AGJUAckQh2QCHVArqgDEkUdkCjqgEyoA4ZS\n+yA3M6te17WZW9P8QYkLgQuAvweu3LWP18itVl4jX+ntaz13epmbXf9C0HeBHwNOT67zQeBRk/dm\nZia0yNLK6cn7C4DzgW/2n9O7UAckQh2QCHVAItQBuaIOSBR1QKKoAzKhDhjKIoN8AzgBfA04Bnxy\nkCIzM1vIIn98eRu4Argd8C6az25l1z5XAycnp0/RDP7pPjF5P/Z55lzu883bKvUws21FevbLXOSw\nfe0/3RbkH4r77b/o8ZfZf/d1hu5Zm3kwPb1Fj5ZdZH8J8B3gFTPb/M1Oq5W/2bnS29d67oz6gqDL\ngEsnp28D/CRw/KA3PoJQByRCHZAIdUAi1AG5og5IFHVAoqgDMqEOGErXpZW7AH9NM/g3gNcC/zhU\nlJmZdefftWI2n5dWVnr7Ws8d/64VMzOrf5CHOiAR6oBEqAMSoQ7IFXVAoqgDEkUdkAl1wFBqH+Rm\nZtXzGrnZfF4jX+ntaz13vEZuZmb1D/JQByRCHZAIdUAi1AG5og5IFHVAoqgDMqEOGErtg9zMrHpe\nIzebz2vkK719reeO18jNzKz+QR7qgESoAxKhDkiEOiBX1AGJog5IFHVAJtQBQ6l9kJuZVc9r5Gbz\neY18pbev9dzxGrmZmdU/yEMdkAh1QCLUAYlQB+SKOiBR1AGJog7IhDpgKLUPcjOz6nmN3Gw+r5Gv\n9Pa1njujrpFfDhwDPgFcDzzvoDdsZmbjujNwxeT0bYFPAw/YtU/bUwqlUAckQh2QCHVAItQBM3Zg\nZ/J2bOb07PadAbZ3vc6YTV23HxvxdjuLnh4PfeplbnZ9Rv6fwInJ6f8BbgDu2keAmZkdzDJrM1vA\n+4AH0gz1Ka+RW628Rr7S29d67vQyNzcX3P+2wJuA53PuEJ+6Gjg5OX2K5ll8mZyPyXuf9/mez2/e\nBGcuZo+N07B9Uffjbbwdto/sPc5+V5t3+VD7T7cd9v03gTMtnwUX/f8/yvnp6a3dtWO5FfAu4AUt\nl3uNvJtQByRCHZCIBfbtY/10znG8Rr7Y9jHXyPe775Z+TI2ll7nZdY38POCvgE8CV/Vxw2Zm1o+u\nazOPAt4PXMvZzyBXAu+c2WdngeOZ9Wmnp/XTfY6j2K687XXfPr1s5WdSL3PTLwiyGniQe3vLZSs/\nk3qZm7W/RD/UAYlQByRCHZAIdUCuqAMSRR2QKOqATKgDhlL7IDczq56XVqwGXlrx9pbLVn4meWnF\nzMzqH+ShDkiEOiAR6oBEqANyRR2QKOqARFEHZEIdMJTaB7mZWfW8Rm418Bq5t7dctvIzyWvkZmZW\n/yAPdUAi1AGJUAckQh2QK+qARFEHJIo6IBPqgKHUPsjNzKrnNXKrgdfIvb3lspWfSV4jNzOz+gd5\nqAMSoQ5IhDogEeqAXFEHJIo6IFHUAZlQBwyl9kFuZlY9r5FbDbxG7u0tl638TPIauZmZdR/krwK+\nBlw3YMsQQh2QCHVAItQBiVAH5Io6IFHUAYmiDsiEOmAoXQf5q4HHDRliZmbLWWRtZgt4K/Cglsu9\nRm4qXiP39pbLVn4m9TI3N3sImXW0ZfurgC/0fFtmZkbvg/yeL4N7TE7fGrgz8JH/hRs+x9lBXibv\nY4TzVwBXjXh7Xc5Pt61Kz2xLh/03r4EzF7PHxmnYfkJ+/c2bWq4DbO/dzObNcOalwImO/fvsdosO\nPbP2O14kx2/bv63noPtPt2Ut8/Zf9PjL7H8VzYefqmf35bPzoPXxeBq2L2o5UF/np6e39t7+OLbY\n/5udO/D2HdjZ9fbzNwPPHCdxjxDd7n5CHZCIBfZN/h/v7ND+9e2c67Qea8imJXqmp48tuP9Btq9i\nU9ftx0S3u+///+ljquv+Y+jlNmv/8cOiDkgUdUCiqAMSRR2QC3VAItQBiVAHZIo6YChdB/kbgA8D\n9wW+CDx7sCIzM1tI10H+NOCuwIXA5TQ/jrgOQh2QCHVAItQBiVAH5Io6IFHUAYmiDsiEOmAotS+t\nmJlVr/ZBXtQBiaIOSBR1QKKoA3KhDkiEOiAR6oBMUQcMpfZBbmZWvdoHeagDEqEOSIQ6IBHqgFxR\nBySKOiBR1AGZUAcMpfZBbmZWvdoHeVEHJIo6IFHUAYmiDsiFOiAR6oBEqAMyRR0wlNoHuZlZ9Wof\n5KEOSIQ6IBHqgESoA3JFHZAo6oBEUQdkQh0wlNoHuZlZ9Wof5EUdkCjqgERRBySKOiAX6oBEqAMS\noQ7IFHXAUGof5GZm1at9kIc6IBHqgESoAxKhDsgVdUCiqAMSRR2QCXXAUGof5GZm1at9kBd1QKKo\nAxJFHZAo6oBcqAMSoQ5IhDogU9QBQ6l9kJuZVW+RQf444FPAZ4AXDZPTu1AHJEIdkAh1QCLUAbmi\nDkgUdUCiqAMyoQ4YStdBfj7wxzTD/Ado/tDEA4aK6tEV83cZnZu6WcUmmr8HvWrc1NGKPqYOrusg\nfzjwWeAk8D3gb4CnDNTUp0vVAQk3dbOKTcApdUDCTR2t6GPq4LoO8rvR/K3OqS9NtpmZmdhmx/12\nuu32otPwiu+du+3aC4HtRaJ6tCW63f1sqQMSW+qAxJY6IHdSHZA4qQ5InFQHZLbUAUM5r+N+jwCO\n0qyRA1xJM5xfPrPPZ4F791ZmZla/G4H7jHVjm5Mb3AIuoPlOxjp8s9PMzGb8NPBpmmfeV4pbzMzM\nzMwOj3kvBHoK8HHgOPDvwI/PXHYSuHZy2UdGbJp6GHAG+Nklrjt210k091UA35rc7nHgdxa47lhN\nL5m57CS6x1RMbvd6zn3Fi+p+2q/pJMPcT126XsjZ/3fX0TzWL+14XUXTSTSPqcuAd9IsVV8P/PIC\n113I+TRLKVvArcjXxi+aOf2gyf5Tnwduf9CIJZqm+/0T8DbODsyu1x27C3T3VQDXLHndsZtAdz9d\nCnwCuPvk/GULXHfsJhjmfuraNeuJwHuXvO4YTaB7TB0Ffm9y+jLgGzTfj1z4fpr3c+RdXgj07ZnT\ntwX+e9flXX8ypquuL076deBNwNeXuO7YXVOq+yq73aHuq4M0dblsqKanA39H8xoKOPs4V95PbU1T\nfd9PXbtmPR14w5LXHaNpSvGY+ipwyeT0JTSD/EzH655j3iDv+kKgnwFuAP4BeN7M9h2az3wfBZ4z\n57a66tJ0N5p/+J/NdHS9rqJrelpxX+0Aj6RZHnsHza9g6HrdsZumlynup++nedZ2bHLbz1jgumM3\nwTD3U9euqSPAT9F8sln0umM1ge4x9ZfAA4Gv0DzWn7/Adc8x7wVBHV8IxFsmb48GXgvcb7L9h2k+\n69wReA/Nms8HOh7zIE1XAS+e7HseZz/bdv33LOMgXaC7rz4GXA6cpvnJpLcA9z3g7Q7ZpLqfbgU8\nBPgJmmHwz8C/dLzu2E2fAR5FMyD6vJ+6dk09CfggZ1+vr7yvpnY3ge4x9ds0yyZB8xqc9wAPXubG\n5j0j/zLNB9TU5Zz9Mi7zAZpPDneYnP/q5P3XgTfTfMlwUF2aHkrz5cjnadah/xR4csfrKrpAd1/d\nTDMwofmK6lY0z/K+1OG6YzeB7n76IvBu4Ds0XwK/n+aDbqjH1EGaoBni0O/91LVr6hc5dwlDeV+1\nNYHuMfVI4I2T0zfSzIX7McDHXpcXAt2bs88sHzLZH5pnCBdPTl8EfAh47EFiFmia9WrgqUted6wu\n5X11J87+/3s4Z19bPdR9dZAm5f10f5ovv8+fdFxHs+SjvJ/amoa6n7p2AdyO5pPLbZa47phNysfU\nHwAvm5y+E82wvn3H6y4seyHQr0zeAH6L5kdnjtM8I3/YZPu9JgHTH63p80VE85pmzQ7Mtuuqu5T3\n1a9ObvME8GGaX8ew33WVTerH1AtpfkrkOs79XpDqfmprGvJ+6tr1LOD1Ha+rbLonusfUZcBbadbH\nr6P5Jux+1zUzMzMzMzMzMzMzMzMzMzMzMzMzMzMzMzOzGvw/ugXvS81TePoAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f8a8c4d3e50>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('metaBEAT_read_stats.csv',index_col=0)\n",
    "df['fraction'] = df['queries']/(df['total']*0.5)\n",
    "df.fraction.hist(bins=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\n"
     ]
    }
   ],
   "source": [
    "cd ../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Next stage of the processing is chimera detection and removal of putative chimeric sequences\n",
    "\n",
    "We'll do that using `uchime` as implemented in `vsearch`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir chimera_detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/chimera_detection\n"
     ]
    }
   ],
   "source": [
    "cd chimera_detection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Convert reference database from Genbank to fasta format to be used in chimera detection.\n",
    "\n",
    "Prepare Refmap file, i.e. text file that specifies the location and the format of the reference to be used.\n",
    "\n",
    "The reference sequences in Genbank format should already be present in the `supplementary_data` directory: `12S_Fish_SATIVA_cleaned_May_2017.gb`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Write REFmap\n",
    "for file in $(ls -1 ../supplementary_data/reference_DBs/* | grep \"12S_Fish_SATIVA_cleaned_May_2017.gb$\")\n",
    "do\n",
    "      echo -e \"$file\\tgb\"\n",
    "done > REFmap.txt\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../supplementary_data/reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb\tgb\r\n"
     ]
    }
   ],
   "source": [
    "!cat REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\n",
      "version: v.0.97.8-global\n",
      "\n",
      "\n",
      "Thu Nov  2 13:08:40 2017\n",
      "\n",
      "/usr/bin/metaBEAT_global.py -R REFmap.txt -f\n",
      "\n",
      "\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\n",
      "\n",
      "As the mail address is not specified in the script itself (variable 'Entrez.email'), metaBEAT expects a simple text file called 'user_email.txt' that contains your email address (first line of file) in the same location as the metaBEAT.py script (in your case: /usr/bin/)\n",
      "\n",
      "found 'haikuilee@gmail.com' in /usr/bin/user_email.txt\n",
      "\n",
      "taxonomy.db found at /usr/bin/taxonomy.db\n",
      "\n",
      "######## PROCESSING REFERENCE DATA ########\n",
      "\n",
      "\n",
      "processing ../supplementary_data/reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb (containing 340 records)\n",
      "\n",
      "total number of valid records: 340\n",
      "\n",
      "write out reference sequences to refs.fasta\n",
      "\n",
      "\n",
      "Thu Nov  2 13:08:40 2017\n",
      "\n",
      "\n",
      "### GLOBAL CLUSTERING ###\n",
      "\n",
      "\n",
      "Concatenating for global clustering\n",
      "\n",
      "\n",
      "Performing global clustering\n",
      "\n",
      "vsearch --cluster_fast /home/working/chimera_detection/GLOBAL/global_queries.fasta --id 1.00 --strand both --threads 1 --centroids global_centroids.fasta --uc global.uc\n",
      "vsearch v1.1.0_linux_x86_64, 62.8GB RAM, 16 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "\n",
      "\n",
      "parse vsearch uc file\n",
      "\n",
      "\n",
      "writing denovo OTU table\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/usr/bin/metaBEAT_global.py\", line 2448, in <module>\n",
      "    out.write(BIOM_tables_per_method['OTU_denovo'].to_tsv()) #to_json('generaged by test', direct_io=out)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 4027, in to_tsv\n",
      "    observation_column_name)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 1268, in delimited_self\n",
      "    raise TableException(\"Cannot delimit self if I don't have data...\")\n",
      "biom.exception.TableException: Cannot delimit self if I don't have data...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-R REFmap.txt \\\n",
    "-f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">KC292935|7962|Cyprinus carpio\r\n",
      "CCAGAAAAGAGAGATTTTAACTCTCACCCCTGGCTCCCAAAGCCAGAATTCTAAACTAAACTATTTTCTGGGGAGGTAGCACTCCCTTTATGGTATAGTACATATTATGCATAATATTACATTAATGTATTAGTACATATATGTATTATCACCAACTCACTATTTTAACCATAAAGCAGGTACATAATATTAAGGTGGGCATAAAGCATATTATTAAGACTCACAAATTCTATTATTTGGACTTGAGTAATATATTAATCCCCAAAAATTTGTCCTCAAATTTTTCCTTGAAATAATCAACTATAATTCCATTTAAACATATTAATGTAGTAAGAGACCACCAACCAGTTTATATAAAGGTATATCATGAATGATAGAATCAAGGACAATAATTGTGAGGGTTACACAATATGAACTATTACTGGCATCTGGTTCCTATTTCAGGGACATAACTGTAATATTCCACCCTCGGATAATTATACTGGCATCTGATTAATGGTGTAAGTACATACGTTTCATTACCCCCCATGCCGAGCATTCTTTTATATGCATAAGGTATTTTTTATTGGTTTCCTTTCATCTGGCATTTCAGAGTGCAGGCTCAAATGTTAAATCAAGGTTGAACATTTTCCTTGTATGTGATAATATATATTAATTATCGTAAGACATAATTTAAGAATTACATACTTTTATCTCAAGTGCATAATATATCTGTCTCTAGTTCAACTTATCCTTACATAGTGCCCCCTTTGGTTTTTGCGCGACAAACCCCCTTACCCCCTACGCTCAGCGAATCCTGTTATCCTTGTCAAACCCCGAAACCAAGGAGGACCCAAGAACGTGTAAACCAACGAGTTGAGGTATAAATTGGCATCCCATTATATATATATATATATATGCATCGGTTTTTTTAACCGCAACTTACCACTTACCTAAAAGCCCCTACCAAAAATCCCCAAAAAGAGGCTCGGCACTAAATACTCTAATATAATTAATCAGCTAGCGTAGCTTAACACAAAGCATAGCACTGAAGATGCTAAGATGAGTCCTAAAAAACTCCGCATGCACAAAGGCATGGTCCCGACCTTATTATCAGCTCTAACTCAACTTACACATGCAAGTCTCCGCAACCCAGTGAATATGCCCTCAATCCCCCCACCCGGGGACGAGGAGCGGGCATCAGGCACAAACATTAGCCCAAGACGCCTAGCCAAGCCACACCCCCAAGGGAATTCAGCAGTGATAAACATTAAGCCATAAGTGAAAACTTGACTCAGTTAGTGTTAAGAGGGCCGGTAAAACTCGTGCCAGCCACCGCGGTTAGACGAGAGGCCCTAGTTGATATTACAACGGCGTAAAGGGTGGTTAAGGATAAACAAAAATAAAGTCAAATGGCCCCTTGGCCGTCATACGCTTCTAGGAGTCCGAAGCCCTAATACGAAAGTAACTTAATAAACCCACC\r\n",
      ">AJ001798|98395|Cobitis taenia\r\n",
      "TACCCCACTATGTTCAGCCATAAACTTAGACGTTGTTATACAATAAACGTCCGCCAGGGTACTACGAGCATTAGCTTAAAACCCAAAGGACTTGGCGGTGCCTCAGACCCCCCTAGAGGAGCCTGTTCTAGAACCGATAACCCCCGTTAAACCTCACCACTTCTGGTCATCCCCGCCTATATACCGCCGTCGTCAGCTTACCCTGTGAAGGACTGATAGTAAGCAAAATGAATAAAATCCAAAACGTCAGGTCGAGGTGTAGCGAACGAAGTGGGAAGAAATGGGCTACATTTTCTTTAATAAGAATAATACGAACAACACTATGAAAACTAGTGCTCGAAGGAGGATTTAGTAGTAAAAAGGAAATAGAGTGTCCTTTTGAACCCGGCTCTGAG\r\n",
      ">KF856964|7962|Cyprinus carpio\r\n",
      "GCTAGCGTAGCTTAACACAAAGCATAGCACTGAAGATGCTAAGATGAGTCCTAAAAAACTCCGCATGCACAAAGGCATGGTCCCGACCTTATTATCAGCTCTAACTCAACTTACACATGCAAGTCTCCGCAACCCAGTGAATATGCCCTCAATCCCCCCACCCGGGGACGAGGAGCGGGCATCAGGCACAAACATTAGCCCAAGACGCCTAGCCAAGCCACACCCCCAAGGGAATTCAGCAGTGATAAACATTAAGCCATAAGTGAAAACTTGACCCAGTTAGTGTTAAGAGGGCCGGTAAAACTCGTGCCAGCCACCGCGGTTAGACGAGAGGCCCTAGTTGATATTACAACGGCGTAAAGGGTGGTTAAGGATAAACAAAAATAAAGTCAAATGGCCCCTTGGCCGTCATACGCTTCTAGGAGTCCGAAGCCCTAATACGAAAGTAACTTTAATAAACCCACCTGACCCCACGAAAGCTGAGAAACAAACTGGGATTAGATACCCCACTATGCTCAGCCGTAAACTCAGACATCCAGCTACAATTAGATGTCCGCCAGGGTACTACGAGCATTAGCTTAAAACCCAAAGGACCTGACGGTGTCTCAGACCCCCCTAGAGGAGCCTGTTCTAGAACCGATAACCCCCGTTCAACCTCACCACTTCTAGCCACCCCAGCCTATATACCGCCGTCGTCAGCTTACCCTGTGAAGGTAATAAAAGTAAGCAAAATGGGCACAACCCAAAACGTCAGGTCGAGGTGTAGCGCATGAAGTGGGAAGAAATGGGCTACATTTTCTAATATAGAATATTACGAACATGCACCATGAAACAATGCTTGAAGGAGGATTTAGTAGTAAAAGGGAAATAGAGTGTCCCTTTGAACCCGGCTCTGAGACGCGTACACACCGCCCGTCACTCTCCCCTGTCAAAACGCACCAAAAATACATAATACAACAGCACTGACAAGGGGAGGCAAGTCGTAACACGGTAAGTGTACCGGAAGGTGCACTTGGATCAAACCCAGGGTGTGGCTGAGTTAGTCAAGCATCTCACTTACACCGAGAAGACATCCATGCAAGTTGGATCGCCCTGAGCCAAACAGCTAGCTTAACTACTTAATAACTAAACAATATAAATAAAACAAAATAAGCCTAACACCAAAAACTAAACCATTTTTTTACCTGAGTATGGGAGACAGAAAAGGTTCCACAAAGCGATAGAAATAGTACCGCAAGGGAAAGCTGAAAGAGAAATGAAATAACCCATATAAGCACTAAAAAGCAAAGATTAAACCTCGTACCTTTTGCATCATGATTTAGCCAGTACACCCAAGCAAAGAGACCTTTAGTTTGAAACCCCGAAACCAGGTGAGCTACCCCGAGACAGCCTATTAAGGGCCAACCCGTCTCTGTGGCAAAAGAGTGGGAAGAGCTCCGGGTAGAAGTGACAGACCTACCGAACCTGGTGATAGCTGGTTGCCTAAGAAATGAATAGAAGTTCAGCCTCGTGCACCTCAAATCACACAAACAAAACAAGACTAAGAGAAACACATGAGAGTTAGTTAAAGGGGGTACAGCCCCTTTAACAAAGGACACAACCTTTCCAGGAGGATAAAGATCATAATACATAAAACATACTGTTCTAGTGGGCCTAAAAGCAGCCATCTAAACAGAAAGCGTTAAAGCTCAGACAGAAAAAAGTTTATTATCCCGATAAAAAATCTTACTCCCCTAAGTACTATTAGGCCAACCCATGCCCCCATGGAAGAGATTATGCTAAAATGAGTAACAAGAAGGCCCGCCCTTCTCCAAGCACAAGTGTAAGCCAAACCGGACCAACCATTGGCAACTAACGAACTCAATCAAAGAGAGCAATGTGAATTACAAAAAAACCAAGAAAAACCCACAACTAAGCTATCGTTACCCCCACACTGGAGTGCCCTAAAGGAAAGACTAAAAGAAAAGGAAGGAACTCGGCAAACACAAGCCTCGCCTGTTTACCAAAAACATCGCCTCCTGCAACGCAACCAAGTATAGGAGGTCCAGCCTGCCCAGTGACTACAAGTTCAACGGCCGCGGTATTTTGACCGTGCAAAGGTAGCGCAATCACTTGTCTTTTAAATAGAGACCTGTATGAATGGCTAAACGAGGGCTTAACTGTCTCCCCTTTCAAGTCAGTGAAATTGATCTACCCGTGCAGAAGCGGGTATAATACTACAAGACGAGAAGACCCTTTGGAGCTTAAGGTACAAAACTCAACCACGTTAAGCAACTCAATAAAAAGCAAAAACCTTGTGGACCATGAGATTTTACCTTCGGTTGGGGCGACCACGGAGGAAAGAAAAGCCTCCAGGTGGACTGGGAAAACCTCCTAAAACCAAGAGAGACATCTCTAAGCCACAGAACATCTGACCAAATATGATCCGGCTAACACATAGCCGATCAACGAACCAAGTTACCCTAGGGATAACAGCGCAATCCTCTCCCAGAGTCCATATCGACGAGGGGGTTTACGACCTCGATGTTGGATCAGGACATCCTAATGGTGCAGCCGCTATTAAGGGTTCGTTTGTTCAACGATTAAAGTCCTACGTGATCTGAGTTCAGACCGGAGCAATCCAGGTCAGTTTCTATCTGTAACGCTACTTTTCCTAGTACGAAAGGATCGGAAAAGAGGGGCCCATACTTAAAGCACGCCCCACCCCTAATTTATGAAAACAAATAAATAAAATAAAGGGAGAGCCAAAATCCCAGCTGGCCAAAATAAGGACATACTGGGGTGGCAGAGCATGGTAAATTGCGAAAGGCCTAAGCCCTTTTACCCAGAGGTTCAAATCCTCTTCCCAGTTTATGCTAAACATCCTAATAACTCACCTAATCAACCCTCTAGCCTATATCGTGCCTGTACTTCTAGCAGTAGCCTTCCTAACACTAATCGAACGAAAAGTACTAGGATATATGCAACTACGAAAAGGACCAAACGTGGTAGGCCCCTACGGACTACTACAACCCATCGCCGACGGAGTAAAACTCTTCATTAAAGAACCCGTCCGCCCATCCACATCATCCCCATTCCTATTCCTAGCCGCCCCAGTACTTGCATTAACCCTAGCCATAACCTTATGAGCACCAATACCTATACCCCACCCAGTAACTGACCTCAACCTAGGAATCCTATTTATTCTAGCCCTATCAAGCCTTGCGGTATACTCAATCCTAGGGTCAGGATGAGCATCAAATTCAAAATATGCATTAATCGGAGCCCTACGGGCCGTAGCCCAAACAATTTCGTATGAAGTTAGCCTAGGACTAATCCTCTTATCCGTAATTATCTTCTCAGGAGGATATACCCTACAAACATTTAACATCACCCAAGAAAGCATTTGACTTCTCATCCCTGCCTGACCTTTAGCCGCAATATGATATATCTCAACATTAGCCGAAACAAACCGAGCACCATTCGACCTAACAGAAGGAGAATCAGAACTGGTATCCGGCTTCAACGTAGAATATGCAGGAGGACCCTTCGCACTCTTCTTCCTAGCCGAATACGCCAACATCCTTCTAATAAATACCCTCTCAGCCGTCCTATTTTTAGGAGCCTCACATATTCCAAGCATACCAGAACTTACGACAATTAATCTCATAACTAAAGCCGCACTACTCTCTATTATATTCCTATGAGTACGAGCCTCGTACCCACGATTCCGATACGACCAACTAATACACTTAGTATGAAAAAACTTCCTTCCCCTCACACTCGCCTTTGTACTATGACACACTGCCCTACCAATTGCACTAGCAGGGCTCCCTCCACAACTATAATTAAGGAACTGTGCCTGAGCACCTAAGGACCACTTTGATAGAGTGAATTACAGGGGTTAAAATCCCCTCAGTTCCTAGAAAGAAGGGAATTGAACCCATACTCAAGAGATCAAAACTCTTGGTGCTTCCTTTACACCACTTTCTAAAGGCGGGGTCAGCTAATAAAGCTTTCGGGCCCATACCCCGAACATGACGGTTAAAGTCCCTCCTCCGCCAATGAACCCATATGTACTCGCAACCCTATTATCCAGCCTAGGGCTAGGAACCACCCTAACCTTTGCTAGCTCTCACTGACTCCTAGCTTGAATGGGCCTAGAAATTAATACGCTAGCAATCACCCCCCTAATGGCACAACATCACCACCCCCGTGCAGTAGAAGCAACTACAAAATATTTCTTAACCCAAGCCACCGCCGCGGCAATAATCCTGTTCGCAAGCACAACAAATGCCTGAATAACAGGAGAATGAAGCATCAATGACCTGTCGAACCCCATCGCCAGCACAATATTTATAGCTGCCTTAGCACTTAAAATTGGACTTGCACCAATACACTTCTGAATACCAGAGGTTCTACAAGGATTAGACCTGCTAACAGGACTCATCCTATCTACCTGACAAAAACTCGCCCCATTCGCGCTTATTATCCAAACAGCACAAACCATCGACCCACTGCTACTAACACTATTAGGAATTTCATCCACATTAGTTGGAGGATGAGGGGGATTAAATCAAACCCAACTACGAAAAATCCTAGCCTACTCTTCAATCGCACACATAGGATGAATAATTATTGTAATCCAATACGCCCCACAACTCACCCTAATTGCACTAGGAACATATATTATCATAACCTCCGCAGCATTCCTGACCCTAAAAATATCACTAACAACCAAAATCAGCACACTCGCAACAACCTGATCAAAAAGCCCTATCCTGACATCAACAACTGCCCTAGTCCTACTCTCACTAGGAGGCCTACCACCATTAACAGGATTTATACCAAAATGACTAATTCTACAAGAACTAACAAAACAAGACCTCCCAATTATTGCCACAGCCATAGCCCTAACCGCCCTAATTAGCCTATACTTCTACCTACGCCTATGCTACGCAATAACACTGACTATTTCCCCCAATATGATTAACTCAACAACCCCCTGACGAACCCAAACAACCCAAACCTCTCTACCCTTAGCCCTATTTACCACAGCCGCCCTCGGACTCCTACCAATAGCCCCAGCTATCCTAATACTAACCACCTAGGGATTTAGGATAACATTAGACCAAAAGCCTTCAAAGCTTTAAGTAGAAGTGAAAATCTTCTAATCCCTGATTAAGGCCTACAAGATATTAACTTGCATCTCCTGATTGCAAATCAGACACTTTTATTAAGCTAAGGCCTTACTAGATGGGAAGGCCTCGATCCTACAAACTTTTAGTTAACAGCTAAACGCTCAAGCCAGCGAGCATCCATCTACTTTTCCCGCCGTTTAACTCAGTAAGGCGGGAAAAGCCCCGGCAGAGTATTAATCTGCGTCTTCGGATTTGCAATCCAATGTGTTCTTCACCACGGAGCTGATAGGAAGAGGACTCAAACCTCTGTCTTCGGGGCTACAACCCACCGCCTAAGCACTCGGCTACCCTACCTGTGGCAATCACGCGCTGATTCTTCTCTACCAACCACAAAGACATTGGCACCCTTTATCTTGTATTTGGTGCCTGAGCCGGAATAGTAGGAACCGCCTTAAGCCTCCTCATTCGGGCCGAACTTAGCCAACCCGGGTCGCTTCTAGGTGATGACCAAATTTATAACGTTATCGTCACTGCCCACGCCTTTGTAATAATTTTCTTTATAGTAATGCCTATCCTTATTGGAGGATTTGGAAACTGACTTGTACCACTAATAATCGGAGCCCCAGACATAGCATTCCCACGAATAAATAACATAAGCTTCTGACTACTACCCCCATCATTCCTTCTACTCCTAGCTTCTTCTGGTGTTGAAGCTGGAGCTGGAACAGGATGAACCGTATACCCACCTCTTGCAGGGAACTTAGCCCACGCAGGAGCATCAGTAGACCTAACAATTTTCTCACTTCACCTAGCAGGTGTTTCATCAATTCTAGGGGCAATCAACTTTATTACTACAACCATCAACATGAAACCCCCAGCCATCTCTCAATACCAAACACCCCTGTTCGTCTGATCCGTGCTTGTAACCGCCGTATTGCTCCTTCTATCATTACCTGTTTTAGCCGCAGGAATTACAATGCTCCTAACAGACCGAAACCTTAATACCACATTCTTTGACCCGGCAGGAGGAGGAGACCCAATCCTTTATCAACACTTATTCTGATTCTTCGGCCACCCAGAAGTTTATATTCTCATCCTCCCAGGATTCGGTATCATCTCCCATGTTGTAGCCTATTATTCAGGCAAAAAAGAACCGTTCGGTTACATAGGAATGGTCTGAGCCATGATGGCCATCGGTCTACTAGGGTTTATTGTGTGAGCTCACCACATGTTCACTGTTGGAATAGACGTAGACACTCGAGCATATTTTACATCCGCAACAATAATTATCGCAATTCCAACAGGTGTAAAAGTATTTAGCTGATTAGCCACACTCCACGGAGGATCCATTAAATGAGAAACACCAATACTATGGGCTCTAGGGTTTATTTTCCTATTCACAGTAGGGGGACTCACAGGGATCGTCCTATCCAATTCATCACTTGATATTGTCCTTCACGACACTTACTACGTAGTAGCACACTTCCACTATGTACTATCAATGGGTGCTGTATTCGCTATTATGGCAGCCTTTGTGCATTGATTCCCCCTATTAACTGGGTATACTCTCCACAGCACCTGAACAAAAATCCACTTTGGAGTTATGTTTATTGGAGTTAACCTCACGTTCTTCCCACAACACTTCCTAGGCTTAGCCGGTATGCCACGGCGATACTCCGACTATCCAGATGCCTACGCCCTGTGAAATACAGTATCATCCATTGGATCGCTAATTTCCCTAGTAGCAGTAATCATATTCCTATTTATCCTATGAGAAGCCTTCGCCGCTAAACGAGAAGTACTATCTGTAGAACTAACAGCAACAAATGTAGAATGACTTCACGGCTGCCCTCCTCCTTATCACACATACGAGGAACCAGCATTCGTTCAAATTCAATCAAATTAACGAGAAAGGGAGGAATTGAACCCCCATATACTGGTTTCAAGCCAGCCACATAACCACTCTGTCACTTCCTTCTAAAGACATTAGTAAAACGTAAATTACATCACCTTGTCAAGGTGAAATCGTAGGTTAAATCCCTGCATGTCTTAACCCAAAGACTTAATGGCACACCCAACGCAACTAGGATTTCAAGACGCGGCATCACCCGTTATAGAAGAACTTCTTCACTTCCACGACCACGCATTAATAATTGTGCTCCTAATTAGCACTTTAGTGTTATATATTATTACTGCAATGGTATCAACTAAACTTACTAATAAATATATTCTAGACTCCCAAGAAATCGAAATCGTATGAACCATTCTACCAGCCGTCATTTTAGTACTAATCGCCCTGCCCTCCCTACGCATCCTGTACCTTATAGACGAAATTAACGACCCTCACCTGACAATTAAAGCAATAGGACACCAATGATACTGAAGTTACGAGTATACAGACTATGAAAATCTAGGATTCGACTCCTATATAGTACCAACCCAAGACCTTGCCCCCGGACAATTCCGACTTCTGGAAACAGACCACCGAATAGTTGTTCCAATAGAATCCCCAGTCCGTGTCCTAGTATCTGCTGAAGACGTGCTACATTCTTGAGCTGTTCCATCCCTTGGCGTAAAAATGGACGCAGTCCCAGGACGACTAAATCAAGCCGCCTTTATTGCCTCACGCCCAGGGGTGTTTTACGGACAATGCTCTGAAATTTGTGGAGCTAATCACAGCTTTATACCAATTGTAGTTGAAGCAGTACCTCTCGAACACTTCGAAAACTGATCCTCATTAATACTAGAAGACGCCTCGCTAGGAAGCTAAATATTGGACAAAGCGTTGGCCTTTTAAGCCAAAGATTGGTGATTCCCGACCACCTCTAGTGAAATGCCACAATTAAACCCCGGCCCTTGATTCGCAATTTTAGTATTTTCTTGACTAATTTTCTTAACTATTATTCCAACTAAAATCTTAAGCCATATTTCACCAAATGAACCAACCCCAGTAAGTGCTGAAAAACACAAAACTGAATCCTGAGACTGACCATGATAGTAAGCTTTTTCGACCAATTTGCAAGCCCATCATATCTAGGAATTCCCCTAATTGCTATCGCAATCGCACTACCCTGAGTACTTTACCCAACCCCACCAGCTCGATGAATTAATAACCGGCTCATCACGATCCAAGGATGATTTATCAACCGATTTACTAACCAACTAATACTACCACTAAATGTAGGAGGACATAAATGAGCACTACTACTAGCCTCATTAATAATTTTCCTAATTACAATTAATATGCTTGGCTTACTACCGTATACCTTCACACCCACAACACAACTATCACTCAATATGGGATTTGCCGTACCTCTATGACTCGCTACTGTAATTATTGGGATACGAAATCAACCAACAGTTGCCCTAGGACACCTACTACCAGAAGGAACACCCATTCCACTGATCCCAGTACTAATTATCATCGAAACAATTAGCCTATTTATCCGCCCATTAGCCCTGGGAGTCCGACTAACAGCCAATCTAACCGCAGGTCACCTGTTAATCCAACTCATCGCCACAGCTGTATTTGTTCTCCTACCAATAATACCAACAGTAGCAATCCTAACCGCTGCCGTACTCTTCCTCCTCACATTACTAGAAGTTGCAGTAGCAATAATCCAAGCTTATGTATTTGTACTTCTTCTAAGCCTATATCTACAAGAAAACGTTTAATGGCCCACCAAGCACATGCCTATCATATAGTTGATCCAAGCCCATGACCACTAACCGGAGCTATCGCTGCCCTCCTAATAACATCCGGCTTAGCAATCTGATTCCACTTCCACTCAACAACATTAATAACTCTAGGACTAATTCTCCTACTTCTTACCATATATCAATGATGACGTGACATTATCCGAGAAGGGACCTTCCAAGGCCACCACACACCGCCTGTACAAAAAGGGCTACGATATGGAATAATCCTATTTATCACCTCTGAAGTATTCTTCTTCCTAGGATTTTTCTGAGCCTTCTACCACTCAAGCCTAGCACCAACACCAGAATTAGGAGGATGCTGACCTCCCACAGGAATTACACCACTAGATCCCTTCGAAGTACCACTCCTTAATACAGCCGTTCTACTAGCATCAGGAGTCACAGTAACATGAGCTCATCACAGCATCATAGAGGGAGAACGAAAACAAGCTATTCAATCCTTAGCATTAACCATTTTACTAGGACTCTACTTCACCGCCCTCCAAGCCATGGAATACTATGAAGCACCCTTCACAATTGCAGATGGGGTTTACGGTTCAACATTCTTCGTAGCCACAGGTTTCCACGGTCTCCATGTTATTATTGGATCAACCTTCCTAGCAGTATGCCTTCTTCGCCAAATCCAATACCACTTTACATCTGAACACCATTTTGGCTTTGAAGCCGCTGCCTGATACTGACATTTTGTCGACGTAGTATGACTGTTCCTCTACGTATCCATCTATTGATGAGGCTCATAATCTTTCTAGTATCAAAGTTAGTACAAGTGACTTCCAATCACACAGTCTTGGTTAAACCCCAAGGAAAGATAATGAATCTAATTATAACTATTCTAACCATTACCGTGGCCCTATCCCTAATTCTAGCAACCGTATCCTTTTGACTACCACAAATAAATCCAGACGCAGAAAAACTATCACCATACGAATGCGGGTTTGATCCACTAGGATCTGCCCGACTACCATTTTCCCTACGCTTCTTCCTAGTAGCAATCCTATTCCTACTCTTCGATCTAGAAATTGCCCTCCTCCTCCCACTGCCCTGAGGAGACCAACTCCACAACCCCACCGGAACATTCTTCTGGGCCACAACAGTTCTAATTTTATTAACTCTAGGACTAATTTATGAATGAACACAAGGCGGCCTAGAATGAGCAGAATAGGGAGTTAGTCCAAAACAAGACCTCTGATTTCGGCTCAGAAAATCATGGTTTAATTCCATGACCCCCTTATGACACCCGTACATTTCAGCTTTAGCTCAGCATTCATTTTAGGTCTAATGGGACTAGCATTCCACCGAACCCACCTACTCTCCGCGCTACTATGCTTAGAAGGAATAATACTGTCCCTATTCATTGCACTAGCCCTATGAGCATTACAATTTGAATCCACAGGATTCTCAACAGCCCCTATACTACTACTGGCCTTTTCTGCTTGTGAAGCTAGCACCGGCTTAGCACTTTTAGTTGCTACCGCTCGCACTCACGGAACCGACCGGCTACAAAACCTTAATCTCCTACAATGCTAAAAGTACTAATCCCTACAATCATATTATTTCCAACAATTTGATTAACCTCCCCTAAATGACTGTGGACAACTACAACTGCACACAGCCTCTTAATTGCCTCCATCAGCCTAACATGACTGAAATGAACATCCGAAACTGGGTGAACTTCCTCCAACATATACCTAGCCACAGACCCACTATCAACCCCCCTATTAGTCCTAACATGCTGACTACTCCCACTTATAATCCTAGCCAGCCAAAACCATATTAATCCCGAACCAATTAGCCGGCAACGCCTATACATCACACTCCTCGCCTCACTACAAACCTTTCTGATCATAGCATTCGGTGCCACAGAAATCATTATATTCTACATCATATTTGAAGCCACACTCATTCCAACCCTCATTATTATTACCCGCTGAGGGAACCAAACCGAACGCCTTAACGCCGGAACCTACTTCTTATTCTATACTCTAGCAGGATCCCTTCCGCTTTTAGTCGCCCTACTCCTTCTTCAACAATCCACAGGTACACTATCAATACTTGTACTACAATATTCACAACCCCTACAACTCAACTCTTGAGGTCACATGTTTTGATGAGCTGGCTGCCTAATCGCATTCCTAGTCAAGATACCACTATATGGTGTCCACCTGTGATTACCAAAAGCACACGTAGAAGCCCCCGTAGCAGGATCGATAGTACTAGCAGCAGTATTACTAAAACTTGGTGGATACGGAATAATACGTATAATAGTAATACTAGATCCCCTATCAAAAGAACTCGCCTACCCATTTATTATCCTAGCCCTCTGAGGAATCATTATAACCGGATCAATTTGCCTCCGACAAACAGACCTAAAGTCACTAATCGCTTACTCATCCGTTAGCCACATAGGATTAGTAGCAGGAGGAATCCTAATTCAAACCCCATGAGGATTCTCAGGAGCAATCATCCTAATAATTGCTCACGGACTAGTATCCTCGGCACTTTTCTGCTTAGCCAACACAGCCTATGAACGAACCCATAGCCGAACAATAATTCTCGCCCGAGGCCTACAAGTAATTTTCCCACTAACTGCAGTATGATGGTTTATCGCAAATCTAGCTAACTTAGCACTCCCGCCTCTACCCAACCTAATAGGAGAACTTATAATCATCACAACATTATTCAATTGATCCCCCTGAACAATTTTACTCACTGGCCTGGGAACACTAATCACAGCTGGCTACTCCCTATACATGTTCCTCATGTCACAACGGGGCCCAACACCAAACCATATTATAGGCCTTCAACCATTCCACACCCGAGAACACCTACTAATAACTCTACACCTAATCCCTGTTATCCTCTTAGTAACAAAACCGGAACTCATGTGAGGATGATGCTATTGTAAGTATAGTTTAACTAAAACATTAGATTGTGATTCTAAAAACAGGGGTTAAAGTCCCCTTACTCACCAAGGAAGAACAGAAAATAGTAAGCACTGCTAATCCTTACGTTCCATGGTTAAAATCCGTGGCTTCCTTGCGCTTTCAAAGGATAACAGTTCATCCGTTGGTCTTAGGAACCAAAAACTCTTGGTGCAAATCCAAGTGGAAGCTAAAATGACACTAATTATACACTCATCACTCCTTCTAATCTTTTTTATCTTAGTTTACCCCCTACTCACCACATTAAACCCTAACCAGCAAGACTCCAACATAGCAGGTATAACCAAAATCGCCGTTAGCTCTGCATTCTTCGTCAGCCTCCTGCCCCTTATAATTTTCCTGAACCTAAAAACAGAAGGCATTATTACGAACTGACAATGAATAAACACCCAAACATTTGACGTAAACATCAGTTTCAAATTCGACCACTACTCCCTAATCTTCGTCCCAATCGCCCTATACGTCACCTGATCAATTTTAGAATTCGCACTATGATATATACACTCCGACCCCAACATTGACCGATTCTTCAAATACTTACTCACATTCCTAGTAGCCATAATTATCCTAGTTACAGCCAACAACATATTCCAACTATTTATCGGCTGAGAAGGCGTAGGCATCATATCATTCCTACTCATCGGATGATGACATGGACGAGCAGACGCCAACACCGCGGCTCTCCAAGCTGTTATTTACAACCGAGTAGGAGACATTGGACTAATTATAACTATAGCCTGACTAGCAATAAACCTTAATTCCTGAGAAATTCAACAAATCTTTGCCCTGTCAAAAAACTTTGATATAACAATTCCCCTAATAGGACTTGCCCTAGCGGCAACAGGAAAATCAGCCCAATTTGGCCTCCACCCATGACTCCCGTCCGCCATGGAGGGCCCTACACCAGTATCCGCCCTACTCCATTCAAGTACTATAGTCGTCGCAGGAATTTTCCTACTAATCCGCCTTCACCCCCTCATAGAAAACAATCAACTAGCCCTAACAACCTGCCTCTGCCTCGGAGCATTAACCTCACTATTTACAGCCACCTGCGCCCTAACCCAAAACGACATCAAAAAAATTGTAGCCTTCTCAACATCCAGCCAACTAGGACTAATAATAGTTACAATCGGACTAAACCAACCACAACTAGCATTCCTCCACATTTGCACCCATGCCTTCTTCAAGGCAATGCTCTTCCTGTGTTCAGGGTCAATCATCCACAGCCTAAATGACGAACAAGACATCCGAAAAATAGGAGGCCTATTCAACATTATGCCCGCCACCTCAACCTATTTTACAATTGGCAGCCTAGCCTTAACAGGAACCCCCTTCCTAGCAGGATTCTTCTCAAAGGACGCAATTATTGAAGCCCTAAACACCTCCTACCTAAACGCCTGAGCCCTAACCCTAACACTAATCGCTACATCATTCACCGCAGTATACAGTTTTCGACTAGTATACTTTGTAATTATAGGAACCCCACGATTCCTGCCCCTATCTCCAATCAACGAAAATAACCCACTAGTAATTAACTCCATCAAACGACTTGCCTGAGGAAGCATTATTGCAGGACTCATTATTACACAAAACTTCCCACCAATAAAAACACCAATCATAACAATATCAATCACTTTAAAAATAGCAGCCCTCATAGTAACAATTGCAGGCCTGCTAGTAGCCATAGAACTAGCCAATTTAACAAGCAAACAAGTAAAAATCACCCCAATAATCTCCACACACCACTTCTCAAACATATTAGGATTCTACCCTATAATTATCCACCGACTTATTCCAAAACTTAAACTTACTCTAGGACAATCAGCCGCCACCCAACTAGACAAAACATGACTCGAAACCGTTGGGCCGAAAGGTCTAGCACTCACACAAATAGCCATAGCAAAAATTACAAATGACATCACACGAGGAATGATCAAAACATACCTAACCATCTTCCTCCTAACTCTAATCCTGGCCACCATCCCCGTCCTCCTTTAAACCGCACGAAGAGCTCCACGACTCAGGCCCCGAGTAAGCTCCAACACAACAAGTAAGGTTAAAAGCAATACCCAAGCACAAATAACCAACATACCCCCACCAGACGAATACATTACAGCTACCCCACTAATATCCCCACGTAAAACAGAAAATTCCTTTAACCCATCTACAACCACCCATGAACCTTCGTATCAGCCCCCTCAAAAAATCCCTGCCACTAAACCAACTCCTAATAAGTAAACTAAAACATAACCCAACACAGAACGACTCCCCCAAGCTTCCGGGAAAGGCTCGGCGGCCAAAGCTGCCGAATAAGCAAAAACCACAAGCATCCCTCCTAAATAAATTAAGAAAAGAACCAAAGATAAAAAAGAACCTCCATGACCAACCAATACCCCACACCCAACCCCAGCTGCAACCACTAAACCAAGAGCAGCAAAATAAGGCGTAGGATTAGAAGCAACAGCAACTAAGCCCACAACCAAAGCTATTAATAACAAAAACATAAAATAGGTCATAATTCTTGCTCAGACTTTAACCGAGACCAATGACTTGAAGAACCACCGTTGTTATTCAACTACAAGAACCACTAATGGCAAGCCTACGAAAAACACACCCTCTCATTAAAATCGCTAACGACGCACTAGTTGACCTACCAACACCATCCAACATCTCAGCATGATGAAACTTTGGATCCCTCCTAGGACTATGCTTAATTACCCAAATTTTAACCGGCCTATTCCTAGCCATACACTACACCTCAGACATTTCAACCGCATTCTCATCTGTTACCCACATCTGCCGAGACGTAAATTACGGCTGACTAATCCGTAATGTACACGCCAACGGAGCATCATTCTTCTTCATTTGCATTTACATACACATCGCCCGAGGCCTATACTACGGATCATACCTCTACAAAGAAACCTGAAACATTGGTGTAGTCCTTCTACTACTAGTCATGATAACAGCCTTCGTTGGCTATGTTCTTCCATGAGGACAAATATCCTTTTGAGGCGCCACAGTAATCACAAACCTCCTATCTGCCGTACCATACATGGGAGACATGTTAGTCCAATGAATCTGAGGTGGGTTCTCAGTAGACAATGCAACACTAACACGATTCTTCGCATTCCACTTCCTACTACCATTTGTTATTGCCGCCGCAACCATCATCCACCTACTGTTCCTCCACGAAACAGGATCAAACAACCCGATCGGACTAAACTCAGACGCAGACAAAGTCTCTTTCCACCCGTACTTCTCATACAAAGACCTCCTTGGGTTCGTAATTATACTCCTAGCTCTTACACTACTAGCACTATTCTCCCCTAACTTACTAGGAGACCCAGAAAACTTCACCCCCGCAAACCCTCTAGTTACACCACCCCACATCAAACCAGAATGATACTTCCTATTTGCCTACGCCATCCTACGATCAATTCCAAACAAACTCGGAGGTGTCCTTGCACTCCTATTCTCCATTCTGGTATTAATAGTAGTACCACTACTACACACCTCAAAACAACGAGGACTAACATTCCGCCCCATCACCCAATTCCTATTCTGAACCCTAGTAGCGGACATAATTATCCTAACATGAATTGGAGGCATACCAGTAGAGCATCCCTTCATCATTATTGGACAAATTGCATCCGTCCTATACTTCGCACTATTCCTCATTTTTATGCCACTAGCAGGATGGTTAGAAAATAAAGCACTAAAATGAGCTTGCCCTAGTAGCTTAGCCTAAAAGCATCGGTCTTGTAATCCGAAGATCGGAGGTTAAATTCCTCCCTAGCGCCCAGAAAAGAGAGATTTTAACTCTCACCCCTGGCTCCCAAAGCCAGAATTCTAAACTAAACTATTTTCTGGGGAGGTAGCACTCCCTTTATGGTATAGTACATATTATGCATAATATTACATTAATGTATTAGTACATATATGTATTATCACCAACTCACTATTTTAACCATAAAGCAGGTACATAATATTAAGGTGGGCATAAAGCATATTATTAAGACTCACAAATTCTATTATTTGAACTTGAGTAATATATTAATCCCCAAAAATTTGTCCTCAAATTTTTCCTTGAAATAATCAACTATAATTCCATTCAAACATATTAATGTAGTAAGAGACCACCAACCAGTTTATATAAAGGCATATCATGAATGATAGAATCAAGGACAATAATTGTGAGGGTTACACAATATGAACTATTACTGGCATCTGGTTCCTATTTCAGGGACATAACTGTAATATTCCACCCTCGGATAATTATACTGGCATCTGATTAATGGTGTAGTACATACATTTCATTACCCCCCATGCCGAGCATTCTTTTATATGCATAGGGTATTTTTTATTGGTTTCCTTTCATCTGGCATTTCAGAGTGCAGGCTCAAATGTTAAATTAAGGTTGAACATTTTTCTTGTATGTGATAATATATATTAATTATCGTAAGACATAATTTAAGAATTACATACTTTTATCTCAAGTGCATAATATATCTGTCTCTAGTTCAACTTATCCTTACATAGTGCCCCCTTTGGTTTTTGCGCGACAAACCCCCTTACCCCCTACGCTCAGAGAATCCTGTTATCCTTGTCAAACCCCGAAACCAAGGAGGACCCAAGAACGTGTAAACCAACGAGTTGAGGTATAAATTGGCATCCCATTATATATATATATATATATGCATCGCTTTTTTTAACCGCAACTTACCACTTACCTAAAAGTCCCTACCAAAAATCCCCAAAAAGAGGCTCGGCACTAAATACTCTAATATAATTAATCA\r\n",
      ">KJ627918|134920|Pungitius pungitius\r\n",
      "AAACTGGGATTAGATACCCCACTATGCCTAGCCCTAAACATTGATAGAATTTTACAACCTCTATCCGCCCGGGGACTACAAGCATCAGCTTGAAACCCAAAGGACTTGGCGGTGCTTTAGATCCACCTAGAGGAGCCTGTTCTAGAACCGATAACCCCCGTTCAACCTCACCCCTCTTTGTTAATACCGCCTATATACCACCGTCGTCAGCTTACCCTGTGAGGGATCAATAGTAAGCTTAACTGGTACAACCCTAAACGTCAGGTCGAGGTGTAGCGTATGGAGGGGGAAGAAATGGGCTACATTCGCTACAACAGCGAACACGAATGATGTCCTGAAACGTACATCTGAAGGAGGATTTAGCAGTAAGTAGAAAATAGAGTGTTCTACTGAAATTGGCCCTGAAGCGCGCACACACCGCC\r\n",
      ">HM177473|8010|Esox lucius\r\n",
      "TACAATTTTATCCGCCAGGGAACTACAAGCATCAGCTTAAAACCCAAAGGACTTGGCGGTGCTTCAGACCCACCTAGAGGAGCCTGTTCTATAACCGATAACCCCCGCTTAACCTCACCAGCCCTTGTTCACCCCGCCTATATACCACCGTCGCCAGCTTATCCTGTGAAGGTCTCATAATAAGCAAAATGGGTAAAACCCAGAACGTCAGGTCGAGGTGTAGCGCATGAGCTGGGAAGAAATGGGCTACATTTTCTAATTCAGAATATACGAACTGCACTGTGAAACCAGTGCCCAAAGGTGGATTTAGCAGTAAACAGAAAATAGAGTGTTCTCTTGAAACTGGCTCTGAAGCGCGCACACACCGCCCGTCACCCTAA\r\n"
     ]
    }
   ],
   "source": [
    "!head refs.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will produce `refs.fasta`.\n",
    "\n",
    "Now run chimera detection."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "### Detecting chimeras in T1-1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T1-4-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T2-4-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T3-4-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T4-4-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in T5-4-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-1-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-1-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-1-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-1-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-1-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-2-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-2-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-2-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-2-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-2-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-3-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-3-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-3-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-3-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-3-5 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-4-1 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-4-2 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-4-3 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-4-4 ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in TP5-4-5 ###\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "\n",
    "for a in $(cut -f 1 ../trimming/Querymap.txt)\n",
    "do\n",
    "    if [ -s ../trimming/$a/$a\\_trimmed.fasta ]\n",
    "    then\n",
    "        echo -e \"\\n### Detecting chimeras in $a ###\\n\"\n",
    "        mkdir $a\n",
    "        cd $a\n",
    "        vsearch --uchime_ref ../../trimming/$a/$a\\_trimmed.fasta --db ../refs.fasta \\\n",
    "        --nonchimeras $a-nonchimeras.fasta --chimeras $a-chimeras.fasta &> log\n",
    "        cd ..\n",
    "\n",
    "    else\n",
    "        echo -e \"$a is empty\"\n",
    "    fi\n",
    "done\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Last step is taxonomic assignment of reads based on a BLAST - LCA approach using the metaBEAT pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "!mkdir non-chimeras"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/non-chimeras\n"
     ]
    }
   ],
   "source": [
    "cd non-chimeras/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Prepare Querymap and Refmap txt files__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Querymap\n",
    "for a in $(ls -l ../chimera_detection/ | grep \"^d\" | perl -ne 'chomp; @a=split(\" \"); print \"$a[-1]\\n\"')\n",
    "do\n",
    "   if [ \"$a\" != \"GLOBAL\" ]\n",
    "   then\n",
    "      echo -e \"$a-nc\\tfasta\\t../chimera_detection/$a/$a-nonchimeras.fasta\"\n",
    "   fi\n",
    "done > Querymap.txt\n",
    "\n",
    "\n",
    "#Write REFmap\n",
    "for file in $(ls -1 ../supplementary_data/reference_DBs/* | grep \"12S_Fish_SATIVA_cleaned_May_2017.gb$\")\n",
    "do\n",
    "      echo -e \"$file\\tgb\"\n",
    "done > REFmap.txt\n",
    "\n",
    "for file in $(ls -1 ../supplementary_data/reference_DBs/* | grep \"Astatotilapia_calliptera.gb$\")\n",
    "do\n",
    "      echo -e \"$file\\tgb\"      \n",
    "      \n",
    "done >> REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "T1-1-1-nc\tfasta\t../chimera_detection/T1-1-1/T1-1-1-nonchimeras.fasta\r\n",
      "T1-1-2-nc\tfasta\t../chimera_detection/T1-1-2/T1-1-2-nonchimeras.fasta\r\n",
      "T1-1-3-nc\tfasta\t../chimera_detection/T1-1-3/T1-1-3-nonchimeras.fasta\r\n",
      "T1-1-4-nc\tfasta\t../chimera_detection/T1-1-4/T1-1-4-nonchimeras.fasta\r\n",
      "T1-1-5-nc\tfasta\t../chimera_detection/T1-1-5/T1-1-5-nonchimeras.fasta\r\n",
      "T1-2-1-nc\tfasta\t../chimera_detection/T1-2-1/T1-2-1-nonchimeras.fasta\r\n",
      "T1-2-2-nc\tfasta\t../chimera_detection/T1-2-2/T1-2-2-nonchimeras.fasta\r\n",
      "T1-2-3-nc\tfasta\t../chimera_detection/T1-2-3/T1-2-3-nonchimeras.fasta\r\n",
      "T1-2-4-nc\tfasta\t../chimera_detection/T1-2-4/T1-2-4-nonchimeras.fasta\r\n",
      "T1-2-5-nc\tfasta\t../chimera_detection/T1-2-5/T1-2-5-nonchimeras.fasta\r\n",
      "T1-3-1-nc\tfasta\t../chimera_detection/T1-3-1/T1-3-1-nonchimeras.fasta\r\n",
      "T1-3-2-nc\tfasta\t../chimera_detection/T1-3-2/T1-3-2-nonchimeras.fasta\r\n",
      "T1-3-3-nc\tfasta\t../chimera_detection/T1-3-3/T1-3-3-nonchimeras.fasta\r\n",
      "T1-3-4-nc\tfasta\t../chimera_detection/T1-3-4/T1-3-4-nonchimeras.fasta\r\n",
      "T1-3-5-nc\tfasta\t../chimera_detection/T1-3-5/T1-3-5-nonchimeras.fasta\r\n",
      "T1-4-1-nc\tfasta\t../chimera_detection/T1-4-1/T1-4-1-nonchimeras.fasta\r\n",
      "T1-4-2-nc\tfasta\t../chimera_detection/T1-4-2/T1-4-2-nonchimeras.fasta\r\n",
      "T1-4-3-nc\tfasta\t../chimera_detection/T1-4-3/T1-4-3-nonchimeras.fasta\r\n",
      "T1-4-4-nc\tfasta\t../chimera_detection/T1-4-4/T1-4-4-nonchimeras.fasta\r\n",
      "T1-4-5-nc\tfasta\t../chimera_detection/T1-4-5/T1-4-5-nonchimeras.fasta\r\n",
      "T2-1-1-nc\tfasta\t../chimera_detection/T2-1-1/T2-1-1-nonchimeras.fasta\r\n",
      "T2-1-2-nc\tfasta\t../chimera_detection/T2-1-2/T2-1-2-nonchimeras.fasta\r\n",
      "T2-1-3-nc\tfasta\t../chimera_detection/T2-1-3/T2-1-3-nonchimeras.fasta\r\n",
      "T2-1-4-nc\tfasta\t../chimera_detection/T2-1-4/T2-1-4-nonchimeras.fasta\r\n",
      "T2-1-5-nc\tfasta\t../chimera_detection/T2-1-5/T2-1-5-nonchimeras.fasta\r\n",
      "T2-2-1-nc\tfasta\t../chimera_detection/T2-2-1/T2-2-1-nonchimeras.fasta\r\n",
      "T2-2-2-nc\tfasta\t../chimera_detection/T2-2-2/T2-2-2-nonchimeras.fasta\r\n",
      "T2-2-3-nc\tfasta\t../chimera_detection/T2-2-3/T2-2-3-nonchimeras.fasta\r\n",
      "T2-2-4-nc\tfasta\t../chimera_detection/T2-2-4/T2-2-4-nonchimeras.fasta\r\n",
      "T2-2-5-nc\tfasta\t../chimera_detection/T2-2-5/T2-2-5-nonchimeras.fasta\r\n",
      "T2-3-1-nc\tfasta\t../chimera_detection/T2-3-1/T2-3-1-nonchimeras.fasta\r\n",
      "T2-3-2-nc\tfasta\t../chimera_detection/T2-3-2/T2-3-2-nonchimeras.fasta\r\n",
      "T2-3-3-nc\tfasta\t../chimera_detection/T2-3-3/T2-3-3-nonchimeras.fasta\r\n",
      "T2-3-4-nc\tfasta\t../chimera_detection/T2-3-4/T2-3-4-nonchimeras.fasta\r\n",
      "T2-3-5-nc\tfasta\t../chimera_detection/T2-3-5/T2-3-5-nonchimeras.fasta\r\n",
      "T2-4-1-nc\tfasta\t../chimera_detection/T2-4-1/T2-4-1-nonchimeras.fasta\r\n",
      "T2-4-2-nc\tfasta\t../chimera_detection/T2-4-2/T2-4-2-nonchimeras.fasta\r\n",
      "T2-4-3-nc\tfasta\t../chimera_detection/T2-4-3/T2-4-3-nonchimeras.fasta\r\n",
      "T2-4-4-nc\tfasta\t../chimera_detection/T2-4-4/T2-4-4-nonchimeras.fasta\r\n",
      "T2-4-5-nc\tfasta\t../chimera_detection/T2-4-5/T2-4-5-nonchimeras.fasta\r\n",
      "T3-1-1-nc\tfasta\t../chimera_detection/T3-1-1/T3-1-1-nonchimeras.fasta\r\n",
      "T3-1-2-nc\tfasta\t../chimera_detection/T3-1-2/T3-1-2-nonchimeras.fasta\r\n",
      "T3-1-3-nc\tfasta\t../chimera_detection/T3-1-3/T3-1-3-nonchimeras.fasta\r\n",
      "T3-1-4-nc\tfasta\t../chimera_detection/T3-1-4/T3-1-4-nonchimeras.fasta\r\n",
      "T3-1-5-nc\tfasta\t../chimera_detection/T3-1-5/T3-1-5-nonchimeras.fasta\r\n",
      "T3-2-1-nc\tfasta\t../chimera_detection/T3-2-1/T3-2-1-nonchimeras.fasta\r\n",
      "T3-2-2-nc\tfasta\t../chimera_detection/T3-2-2/T3-2-2-nonchimeras.fasta\r\n",
      "T3-2-3-nc\tfasta\t../chimera_detection/T3-2-3/T3-2-3-nonchimeras.fasta\r\n",
      "T3-2-4-nc\tfasta\t../chimera_detection/T3-2-4/T3-2-4-nonchimeras.fasta\r\n",
      "T3-2-5-nc\tfasta\t../chimera_detection/T3-2-5/T3-2-5-nonchimeras.fasta\r\n",
      "T3-3-1-nc\tfasta\t../chimera_detection/T3-3-1/T3-3-1-nonchimeras.fasta\r\n",
      "T3-3-2-nc\tfasta\t../chimera_detection/T3-3-2/T3-3-2-nonchimeras.fasta\r\n",
      "T3-3-3-nc\tfasta\t../chimera_detection/T3-3-3/T3-3-3-nonchimeras.fasta\r\n",
      "T3-3-4-nc\tfasta\t../chimera_detection/T3-3-4/T3-3-4-nonchimeras.fasta\r\n",
      "T3-3-5-nc\tfasta\t../chimera_detection/T3-3-5/T3-3-5-nonchimeras.fasta\r\n",
      "T3-4-1-nc\tfasta\t../chimera_detection/T3-4-1/T3-4-1-nonchimeras.fasta\r\n",
      "T3-4-2-nc\tfasta\t../chimera_detection/T3-4-2/T3-4-2-nonchimeras.fasta\r\n",
      "T3-4-3-nc\tfasta\t../chimera_detection/T3-4-3/T3-4-3-nonchimeras.fasta\r\n",
      "T3-4-4-nc\tfasta\t../chimera_detection/T3-4-4/T3-4-4-nonchimeras.fasta\r\n",
      "T3-4-5-nc\tfasta\t../chimera_detection/T3-4-5/T3-4-5-nonchimeras.fasta\r\n",
      "T4-1-1-nc\tfasta\t../chimera_detection/T4-1-1/T4-1-1-nonchimeras.fasta\r\n",
      "T4-1-2-nc\tfasta\t../chimera_detection/T4-1-2/T4-1-2-nonchimeras.fasta\r\n",
      "T4-1-3-nc\tfasta\t../chimera_detection/T4-1-3/T4-1-3-nonchimeras.fasta\r\n",
      "T4-1-4-nc\tfasta\t../chimera_detection/T4-1-4/T4-1-4-nonchimeras.fasta\r\n",
      "T4-1-5-nc\tfasta\t../chimera_detection/T4-1-5/T4-1-5-nonchimeras.fasta\r\n",
      "T4-2-1-nc\tfasta\t../chimera_detection/T4-2-1/T4-2-1-nonchimeras.fasta\r\n",
      "T4-2-2-nc\tfasta\t../chimera_detection/T4-2-2/T4-2-2-nonchimeras.fasta\r\n",
      "T4-2-3-nc\tfasta\t../chimera_detection/T4-2-3/T4-2-3-nonchimeras.fasta\r\n",
      "T4-2-4-nc\tfasta\t../chimera_detection/T4-2-4/T4-2-4-nonchimeras.fasta\r\n",
      "T4-2-5-nc\tfasta\t../chimera_detection/T4-2-5/T4-2-5-nonchimeras.fasta\r\n",
      "T4-3-1-nc\tfasta\t../chimera_detection/T4-3-1/T4-3-1-nonchimeras.fasta\r\n",
      "T4-3-2-nc\tfasta\t../chimera_detection/T4-3-2/T4-3-2-nonchimeras.fasta\r\n",
      "T4-3-3-nc\tfasta\t../chimera_detection/T4-3-3/T4-3-3-nonchimeras.fasta\r\n",
      "T4-3-4-nc\tfasta\t../chimera_detection/T4-3-4/T4-3-4-nonchimeras.fasta\r\n",
      "T4-3-5-nc\tfasta\t../chimera_detection/T4-3-5/T4-3-5-nonchimeras.fasta\r\n",
      "T4-4-1-nc\tfasta\t../chimera_detection/T4-4-1/T4-4-1-nonchimeras.fasta\r\n",
      "T4-4-2-nc\tfasta\t../chimera_detection/T4-4-2/T4-4-2-nonchimeras.fasta\r\n",
      "T4-4-3-nc\tfasta\t../chimera_detection/T4-4-3/T4-4-3-nonchimeras.fasta\r\n",
      "T4-4-4-nc\tfasta\t../chimera_detection/T4-4-4/T4-4-4-nonchimeras.fasta\r\n",
      "T4-4-5-nc\tfasta\t../chimera_detection/T4-4-5/T4-4-5-nonchimeras.fasta\r\n",
      "T5-1-1-nc\tfasta\t../chimera_detection/T5-1-1/T5-1-1-nonchimeras.fasta\r\n",
      "T5-1-2-nc\tfasta\t../chimera_detection/T5-1-2/T5-1-2-nonchimeras.fasta\r\n",
      "T5-1-3-nc\tfasta\t../chimera_detection/T5-1-3/T5-1-3-nonchimeras.fasta\r\n",
      "T5-1-4-nc\tfasta\t../chimera_detection/T5-1-4/T5-1-4-nonchimeras.fasta\r\n",
      "T5-1-5-nc\tfasta\t../chimera_detection/T5-1-5/T5-1-5-nonchimeras.fasta\r\n",
      "T5-2-1-nc\tfasta\t../chimera_detection/T5-2-1/T5-2-1-nonchimeras.fasta\r\n",
      "T5-2-2-nc\tfasta\t../chimera_detection/T5-2-2/T5-2-2-nonchimeras.fasta\r\n",
      "T5-2-3-nc\tfasta\t../chimera_detection/T5-2-3/T5-2-3-nonchimeras.fasta\r\n",
      "T5-2-4-nc\tfasta\t../chimera_detection/T5-2-4/T5-2-4-nonchimeras.fasta\r\n",
      "T5-2-5-nc\tfasta\t../chimera_detection/T5-2-5/T5-2-5-nonchimeras.fasta\r\n",
      "T5-3-1-nc\tfasta\t../chimera_detection/T5-3-1/T5-3-1-nonchimeras.fasta\r\n",
      "T5-3-2-nc\tfasta\t../chimera_detection/T5-3-2/T5-3-2-nonchimeras.fasta\r\n",
      "T5-3-3-nc\tfasta\t../chimera_detection/T5-3-3/T5-3-3-nonchimeras.fasta\r\n",
      "T5-3-4-nc\tfasta\t../chimera_detection/T5-3-4/T5-3-4-nonchimeras.fasta\r\n",
      "T5-3-5-nc\tfasta\t../chimera_detection/T5-3-5/T5-3-5-nonchimeras.fasta\r\n",
      "T5-4-1-nc\tfasta\t../chimera_detection/T5-4-1/T5-4-1-nonchimeras.fasta\r\n",
      "T5-4-2-nc\tfasta\t../chimera_detection/T5-4-2/T5-4-2-nonchimeras.fasta\r\n",
      "T5-4-3-nc\tfasta\t../chimera_detection/T5-4-3/T5-4-3-nonchimeras.fasta\r\n",
      "T5-4-4-nc\tfasta\t../chimera_detection/T5-4-4/T5-4-4-nonchimeras.fasta\r\n",
      "T5-4-5-nc\tfasta\t../chimera_detection/T5-4-5/T5-4-5-nonchimeras.fasta\r\n",
      "TP5-1-1-nc\tfasta\t../chimera_detection/TP5-1-1/TP5-1-1-nonchimeras.fasta\r\n",
      "TP5-1-2-nc\tfasta\t../chimera_detection/TP5-1-2/TP5-1-2-nonchimeras.fasta\r\n",
      "TP5-1-3-nc\tfasta\t../chimera_detection/TP5-1-3/TP5-1-3-nonchimeras.fasta\r\n",
      "TP5-1-4-nc\tfasta\t../chimera_detection/TP5-1-4/TP5-1-4-nonchimeras.fasta\r\n",
      "TP5-1-5-nc\tfasta\t../chimera_detection/TP5-1-5/TP5-1-5-nonchimeras.fasta\r\n",
      "TP5-2-1-nc\tfasta\t../chimera_detection/TP5-2-1/TP5-2-1-nonchimeras.fasta\r\n",
      "TP5-2-2-nc\tfasta\t../chimera_detection/TP5-2-2/TP5-2-2-nonchimeras.fasta\r\n",
      "TP5-2-3-nc\tfasta\t../chimera_detection/TP5-2-3/TP5-2-3-nonchimeras.fasta\r\n",
      "TP5-2-4-nc\tfasta\t../chimera_detection/TP5-2-4/TP5-2-4-nonchimeras.fasta\r\n",
      "TP5-2-5-nc\tfasta\t../chimera_detection/TP5-2-5/TP5-2-5-nonchimeras.fasta\r\n",
      "TP5-3-1-nc\tfasta\t../chimera_detection/TP5-3-1/TP5-3-1-nonchimeras.fasta\r\n",
      "TP5-3-2-nc\tfasta\t../chimera_detection/TP5-3-2/TP5-3-2-nonchimeras.fasta\r\n",
      "TP5-3-3-nc\tfasta\t../chimera_detection/TP5-3-3/TP5-3-3-nonchimeras.fasta\r\n",
      "TP5-3-4-nc\tfasta\t../chimera_detection/TP5-3-4/TP5-3-4-nonchimeras.fasta\r\n",
      "TP5-3-5-nc\tfasta\t../chimera_detection/TP5-3-5/TP5-3-5-nonchimeras.fasta\r\n",
      "TP5-4-1-nc\tfasta\t../chimera_detection/TP5-4-1/TP5-4-1-nonchimeras.fasta\r\n",
      "TP5-4-2-nc\tfasta\t../chimera_detection/TP5-4-2/TP5-4-2-nonchimeras.fasta\r\n",
      "TP5-4-3-nc\tfasta\t../chimera_detection/TP5-4-3/TP5-4-3-nonchimeras.fasta\r\n",
      "TP5-4-4-nc\tfasta\t../chimera_detection/TP5-4-4/TP5-4-4-nonchimeras.fasta\r\n",
      "TP5-4-5-nc\tfasta\t../chimera_detection/TP5-4-5/TP5-4-5-nonchimeras.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!cat Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../supplementary_data/reference_DBs/12S_Fish_SATIVA_cleaned_May_2017.gb\tgb\r\n",
      "../supplementary_data/reference_DBs/Astatotilapia_calliptera.gb\tgb\r\n"
     ]
    }
   ],
   "source": [
    "!cat REFmap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Sequence clustering and taxonomic assignment via metaBEAT__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-Q Querymap.txt \\\n",
    "-R REFmap.txt \\\n",
    "--cluster --clust_match 1 --clust_cov 3 \\\n",
    "--blast --min_ident 1 \\\n",
    "-m 12S -n 5 \\\n",
    "-E -v \\\n",
    "-@ haikuilee@gmail.com \\\n",
    "-o 12S-trim_30-merged-nonchimeras-cluster_1c3-blast-min_ident_1.0 &> log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Final result of taxonomic assignment can be found in the table `12S-trim_30-merged-nonchimeras-cluster_1c3-blast-min_ident_1.0-by-taxonomy-readcounts.blast.tsv` (see also [here](https://github.com/HullUni-bioinformatics/Li_et_al_2017/blob/master/assignment_results/12S-trim_30-merged-nonchimeras-cluster_1c3-blast-min_ident_1.0-by-taxonomy-readcounts.blast.tsv)). \n",
    "\n",
    "metaBEAT also produced the final result in [BIOM](http://biom-format.org/) format (`12S-trim_30-merged-nonchimeras-cluster_1c3-blast-min_ident_1.0-by-taxonomy-readcounts.blast.biom`  see also [here](https://github.com/HullUni-bioinformatics/Li_et_al_2017/blob/master/assignment_results/12S-trim_30-merged-nonchimeras-cluster_1c3-blast-min_ident_1.0-by-taxonomy-readcounts.blast.biom)), which can be analyzed with a number of tools and visually explored e.g. using [Phinch](http://phinch.org/)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
